---
title: "p21094_COVten_Bulk_Analysis"
author: "GKT"
date: '`r format(file.info("p21094_COVten_Bulk_Analysis.Rmd")$ctime, "%Y-%m-%d")`'
output: 
  html_notebook: 
    code_folding: hide
---

```{r setup}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
library(tidyverse)
library(ggforce)
library(grid)
library(ggpubr)
library(yaml)
library(DESeq2)
library(EDASeq)
library(DT)
library(shiny)
library(EnhancedVolcano)
library(tools)
library(ComplexHeatmap)
library(gridExtra)
library(gtable)
library(circlize)
library(readxl)
library(openxlsx)

source("/yerkes-cifs/runs/home/gktharp/gitrepos/Coding_Club/BulkRNASeq/gktharp/GKTDESeq2Extentions.R")

source("/yerkes-cifs/runs/MSigDB/GSEA_enrich_auto_vX4.R")

```

## Samples

```{r samples}

# p20019_config <- list(reference = "Mmul10", alignmentDir = "../S2.7.3a_mmul10-100/")
config <- yaml.load_file("p21094_COVten_Bulk_Analysis.yaml")
# p20019_config

analysis = list(
  config = config
)
analysis$sampleIDs = dir(config$alignmentDir)
analysis$sampleSTARReads <- sapply(analysis$sampleIDs, function(sid) {read_tsv(paste0(
  config$alignmentDir,
  sid,"/",sid,
  config$STARreadSuffix
), col_names = c("gene_id","unstranded_count","sense_count","antisense_count"))},
simplify = FALSE,
USE.NAMES = TRUE)

# analysis$sampleIDs

metaComponents <- str_split(analysis$sampleIDs,config$splitChr,config$splitCnt)
#metaName <- as.name(config$sidMetadata[3])

analysis$sampleTable <- tibble(SampleID=analysis$sampleIDs,
                                          mc1=unlist(lapply(metaComponents,"[",1)),
                                          mc2=unlist(lapply(metaComponents,"[",2)),
                                          mc3=unlist(lapply(metaComponents,"[",3)),
                                          mc4=unlist(lapply(metaComponents,"[",4)),
                                          mc5=unlist(lapply(metaComponents,"[",5)),
                                          mc6=unlist(lapply(metaComponents,"[",6))
                                          )
analysis$sampleTable <- mutate(analysis$sampleTable, gcid = paste(mc1,mc2,sep = "-"), mc5=if_else(str_starts(mc6,"7-"),"BL",mc5), mc6=str_remove(mc6,"7-"),.keep = "unused", .after = SampleID)

# analysis$sampleTable <- mutate(analysis$sampleTable, gcid = paste(mc1,mc2,sep = "-"), mc5=paste0(mc5,if_else(str_starts(mc6,"7-"),"-7","")), mc6=str_remove(mc6,"7-"),.keep = "unused", .after = SampleID)

names(analysis$sampleTable) <- c("SampleID",config$sidMetadata)
analysis$sampleTable$SubjectID <- factor(analysis$sampleTable$SubjectID)
analysis$sampleTable$Source <- factor(analysis$sampleTable$Source)
analysis$sampleTable$Timepoint <- factor(analysis$sampleTable$Timepoint)

metaData = list(pbmc = read.xlsx("Metadata.xlsx", cols = 1:3))
metaData$pbmc <- mutate(metaData$pbmc, SubjectID = Animal.ID, .keep = "unused", .before = Group)
metaData$bal <- read.xlsx("Metadata.xlsx", sheet = "BAL", cols = 1:3)
analysis$untreatedList <- metaData$pbmc %>% filter(Treatment == "Untreated") %>% pull(SubjectID)
analysis$treatedList <- metaData$pbmc %>% filter(Treatment == "IFNant") %>% pull(SubjectID)
metaData$pbmc <- mutate(metaData$pbmc, SubjectInd = if_else(SubjectID %in% analysis$treatedList, paste0("SI",lapply(SubjectID , function(x) which(x == analysis$treatedList))), paste0("SI",lapply(SubjectID, function(x) which(x == analysis$untreatedList)))))

analysis$sampleTable <- left_join(analysis$sampleTable, metaData$pbmc)
analysis$sampleTable$Group <- factor(analysis$sampleTable$Group)
analysis$sampleTable$Treatment <- factor(analysis$sampleTable$Treatment, c("Untreated","IFNant"))
analysis$sampleTable$SubjectInd <- factor(analysis$sampleTable$SubjectInd)

# Phase II Sample metadata
metaData2 = list(pbmc = read.xlsx("IFNant_Metadata_Phase_2.xlsx", cols = 1:3))
metaData2$pbmc <- mutate(metaData2$pbmc, SubjectID = Animal.ID, Group = Group + 3, Treatment = if_else(Treatment == "Treated","IFNant",Treatment), .keep = "unused", .before = Group)
analysis$untreated2List <- metaData2$pbmc %>% filter(Treatment == "Untreated") %>% pull(SubjectID)
analysis$treated2List <- metaData2$pbmc %>% filter(Treatment == "IFNant") %>% pull(SubjectID)
metaData2$pbmc <- mutate(metaData2$pbmc, SubjectInd = if_else(SubjectID %in% analysis$treated2List, paste0("SI",lapply(SubjectID, function(x) which(x == analysis$treated2List) + 6)),paste0("SI",lapply(SubjectID, function(x) which(x == analysis$untreated2List) + 6))))

config$alignmentDir2 <- "/yerkes-cifs/runs/211028_A00945_0160_AHNCNKDSX2/Unaligned_Lane1/p21232_COVten/S2.7.3a_Mmul10_MT246667.1/"
config$splitCnt2 <- 5
config$sidMetadata2 <- c("gcid","SubjectID","Source","Day","Timepoint")
analysis$sample2IDs <- dir(config$alignmentDir2)

analysis$sample2STARReads <- sapply(analysis$sample2IDs, function(sid) {read_tsv(paste0(
  config$alignmentDir2,
  sid,"/",sid,
  config$STARreadSuffix
), col_names = c("gene_id","unstranded_count","sense_count","antisense_count"))},
simplify = FALSE,
USE.NAMES = TRUE)

metaComponents2 <- str_split(analysis$sample2IDs,config$splitChr,config$splitCnt2)
#metaName <- as.name(config$sidMetadata[3])

analysis$sampleTable2 <- tibble(SampleID=analysis$sample2IDs,
                                          mc1=unlist(lapply(metaComponents2,"[",1)),
                                          mc2=unlist(lapply(metaComponents2,"[",2)),
                                          mc3=unlist(lapply(metaComponents2,"[",3)),
                                          mc4=unlist(lapply(metaComponents2,"[",4)),
                                          mc5=unlist(lapply(metaComponents2,"[",5))                                          )
analysis$sampleTable2 <- mutate(analysis$sampleTable2, gcid = paste(mc1,mc2,sep = "-"), .keep = "unused", .after = SampleID)
analysis$sampleTable2 <- mutate(analysis$sampleTable2, Timepoint = if_else(grepl("-",mc5),"BL",str_replace(mc5,"D","Day")))
names(analysis$sampleTable2) <- c("SampleID",config$sidMetadata2)
analysis$sampleTable2$SubjectID <- factor(analysis$sampleTable2$SubjectID)
analysis$sampleTable2$Source <- factor(analysis$sampleTable2$Source)
analysis$sampleTable2$Timepoint <- factor(analysis$sampleTable2$Timepoint)
analysis$sampleTable2 <- left_join(analysis$sampleTable2, metaData2$pbmc)
analysis$sampleTable2$Group <- factor(analysis$sampleTable2$Group)
analysis$sampleTable2$Treatment <- factor(analysis$sampleTable2$Treatment, c("Untreated","IFNant"))
analysis$sampleTable2$SubjectInd <- factor(analysis$sampleTable2$SubjectInd)


```

### Phase III? Uninfected Treated

```{r}

config$alignmentDirUT <- "/yerkes-cifs/runs/211220_A00945_0173_AH3GHCDSX3/Unaligned_Lane1-2/p21258_Tim-211220A/S2.7.3a_Mmul10_MT246667.1/"
config$splitChrUT <- "_"
config$splitCntUT <- 5
config$sidMetadataUT <- c("gcid","SubjectID","Source","Day","Timepoint")
analysis$sampleUTIDs <- dir(config$alignmentDirUT)

metaComponentsUT <- str_split(analysis$sampleUTIDs,config$splitChrUT,config$splitCntUT, simplify = TRUE)
colnames(metaComponentsUT) <- config$sidMetadataUT
analysis$sampleTableUT <- tibble(SampleID=analysis$sampleUTIDs, as_tibble(metaComponentsUT))
analysis$sampleTableUT <- mutate(analysis$sampleTableUT, Timepoint = if_else(grepl("-",Timepoint),"BL",str_replace(Timepoint,"D","Day")))
analysis$sampleTableUT$SubjectID <- factor(analysis$sampleTableUT$SubjectID)
analysis$sampleTableUT <- mutate(analysis$sampleTableUT, SubjectInd = paste0("SI",lapply(analysis$sampleTableUT$SubjectID, function(x) which (x==unique(analysis$sampleTableUT$SubjectID)))))
analysis$sampleTableUT$Source <- factor(analysis$sampleTableUT$Source)
analysis$sampleTableUT$Timepoint <- factor(analysis$sampleTableUT$Timepoint)
analysis$sampleTableUT$Treatment <- factor(c("Uninfected-IFNant"))

analysis$sampleUTSTARReads <- sapply(analysis$sampleUTIDs, function(sid) {read_tsv(paste0(
  config$alignmentDirUT,
  sid,"/",sid,
  config$STARreadSuffix
), col_names = c("gene_id","unstranded_count","sense_count","antisense_count"))},
simplify = FALSE,
USE.NAMES = TRUE)

```


## DDS
```{r dds}

analysis$mapBins <- do.call(cbind,
                                   sapply(analysis$sampleIDs,                                      function(sid){analysis$sampleSTARReads[[sid]][,"unstranded_count"][c(1:4),]}))
rownames(analysis$mapBins) <- analysis$sampleSTARReads[[1]][c(1:4),][["gene_id"]]


analysis$dds <- DESeqDataSetFromMatrix(
  do.call(cbind,sapply(analysis$sampleIDs, function(sid){analysis$sampleSTARReads[[sid]][,"unstranded_count"][-c(1:4),]})),
  analysis$sampleTable,
  design = ~ 1
)
colnames(analysis$dds) <- analysis$sampleIDs
rownames(analysis$dds) <- analysis$sampleSTARReads[[1]][-c(1:4),][["gene_id"]]
analysis$mapBins <- rbind(analysis$mapBins,"N_identified" = colSums2(counts(analysis$dds)))

analysis$dds <- DESeq(analysis$dds, parallel = TRUE)
analysis$vst <- varianceStabilizingTransformation(analysis$dds,
                                                         blind = FALSE,
                                                         fitType = "parametric")

# Phase II samples separated

analysis$mapBins2 <- do.call(cbind,
                                   sapply(analysis$sample2IDs,                                      function(sid){analysis$sample2STARReads[[sid]][,"unstranded_count"][c(1:4),]}))
rownames(analysis$mapBins2) <- analysis$sample2STARReads[[1]][c(1:4),][["gene_id"]]


analysis$dds2 <- DESeqDataSetFromMatrix(
  do.call(cbind,sapply(analysis$sample2IDs, function(sid){analysis$sample2STARReads[[sid]][,"unstranded_count"][-c(1:4),]})),
  analysis$sampleTable2,
  design = ~ 1
)
colnames(analysis$dds2) <- analysis$sample2IDs
rownames(analysis$dds2) <- analysis$sample2STARReads[[1]][-c(1:4),][["gene_id"]]
analysis$mapBins2 <- rbind(analysis$mapBins2,"N_identified" = colSums2(counts(analysis$dds2)))

analysis$dds2 <- DESeq(analysis$dds2, parallel = TRUE)
analysis$vst2 <- varianceStabilizingTransformation(analysis$dds2,
                                                         blind = FALSE,
                                                         fitType = "parametric")

analysis$sampleTable2alt <- analysis$sampleTable2[c(1:24,32:25,40:33,48:41),]
analysis$sampleTable2alt <- filter(analysis$sampleTable2alt, !(SubjectID %in% c("RRo9","RZq10")))
analysis$dds2alt <- DESeqDataSetFromMatrix(
  do.call(cbind,sapply(analysis$sample2IDs[!grepl("RRo9|RZq10",analysis$sample2IDs)], function(sid){analysis$sample2STARReads[[sid]][,"unstranded_count"][-c(1:4),]})),
  analysis$sampleTable2alt,
  design = ~ 1
)
colnames(analysis$dds2alt) <- analysis$sampleTable2alt$SampleID
rownames(analysis$dds2alt) <- analysis$sample2STARReads[[1]][-c(1:4),][["gene_id"]]
analysis$mapBins2 <- rbind(analysis$mapBins2,"N_identified" = colSums2(counts(analysis$dds2)))

analysis$dds2alt <- DESeq(analysis$dds2alt, parallel = TRUE)
analysis$vst2alt <- varianceStabilizingTransformation(analysis$dds2alt,
                                                         blind = FALSE,
                                                         fitType = "parametric")

analysis$ddsp12 <- DESeqDataSetFromMatrix(
  do.call(cbind, sapply(c(analysis$sampleIDs, analysis$sample2IDs[!grepl("RRo9|RZq10",analysis$sample2IDs)]),
                        function(sid){
                          c(analysis$sampleSTARReads, analysis$sample2STARReads)[[sid]][,"unstranded_count"][-c(1:4),]
                        })),
  bind_rows(analysis$sampleTable, analysis$sampleTable2alt),
  design = ~ 1
)
colnames(analysis$ddsp12) <- c(analysis$sampleIDs,analysis$sample2IDs[!grepl("RRo9|RZq10",analysis$sample2IDs)])

analysis$ddsp12 <- DESeq(analysis$ddsp12, parallel = TRUE)
analysis$vstp12 <- varianceStabilizingTransformation(analysis$ddsp12, blind = FALSE, fitType = "parametric")


```

### DDS Phase III? Uninfected Treated

```{r}

analysis$mapBinsUT <- do.call(cbind,
                                   sapply(analysis$sampleUTIDs,                                      function(sid){analysis$sampleUTSTARReads[[sid]][,"unstranded_count"][c(1:4),]}))
rownames(analysis$mapBinsUT) <- analysis$sampleUTSTARReads[[1]][c(1:4),][["gene_id"]]


analysis$ddsUT <- DESeqDataSetFromMatrix(
  do.call(cbind,sapply(analysis$sampleUTIDs, function(sid){analysis$sampleUTSTARReads[[sid]][,"unstranded_count"][-c(1:4),]})),
  analysis$sampleTableUT,
  design = ~ 1
)
colnames(analysis$ddsUT) <- analysis$sampleUTIDs
rownames(analysis$ddsUT) <- analysis$sampleUTSTARReads[[1]][-c(1:4),][["gene_id"]]
analysis$mapBinsUT <- rbind(analysis$mapBinsUT,"N_identified" = colSums2(counts(analysis$ddsUT)))

analysis$ddsUT <- DESeq(analysis$ddsUT, parallel = TRUE)
analysis$vstUT <- varianceStabilizingTransformation(analysis$ddsUT,
                                                         blind = FALSE,
                                                         fitType = "parametric")



```



# Mapping
```{r mapping, fig.height=11, fig.width=15}

rownames_to_column(as_tibble(analysis$mapBins, rownames = NA), var = "map_result") %>%
  pivot_longer(!map_result, names_to = "SampleID", values_to = "count") %>%
  mutate(SampleID = str_sub(SampleID,end = -18)) %>%
  ggplot(aes(x = SampleID, y = count, fill = factor(map_result, levels = c("N_unmapped","N_multimapping","N_noFeature","N_ambiguous","N_identified")))) +
  geom_bar(stat = "identity") +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 8, face = "bold")) +
  guides(fill=guide_legend(title="Map Result")) +
  ggtitle(paste0(analysis$config$analysis," Mapping"))
  
  
rownames_to_column(as_tibble(analysis$mapBins2, rownames = NA), var = "map_result") %>%
  pivot_longer(!map_result, names_to = "SampleID", values_to = "count") %>%
  mutate(SampleID = str_sub(SampleID,end = -18)) %>%
  ggplot(aes(x = SampleID, y = count, fill = factor(map_result, levels = c("N_unmapped","N_multimapping","N_noFeature","N_ambiguous","N_identified")))) +
  geom_bar(stat = "identity") +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 8, face = "bold")) +
  guides(fill=guide_legend(title="Map Result")) +
  ggtitle(paste0(analysis$config$analysis," PhaseII Mapping"))
  
  

```


### Phase III? Uninfected Treated Mapping

```{r}

rownames_to_column(as_tibble(analysis$mapBinsUT, rownames = NA), var = "map_result") %>%
  pivot_longer(!map_result, names_to = "SampleID", values_to = "count") %>%
  mutate(SampleID = str_sub(SampleID,end = -18)) %>%
  ggplot(aes(x = SampleID, y = count, fill = factor(map_result, levels = c("N_unmapped","N_multimapping","N_noFeature","N_ambiguous","N_identified")))) +
  geom_bar(stat = "identity") +
  theme_bw() + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust = 1, size = 8, face = "bold")) +
  guides(fill=guide_legend(title="Map Result")) +
  ggtitle(paste0(analysis$config$analysis," PhaseII Mapping"))
  
```


# RLE
```{r rle, fig.height=11, fig.width=15}
# par(mar=c(18,3,2,0))
# plotRLE(counts(analysis$dds)[rowMins(counts(analysis$dds))>0,], outline = FALSE, main = "RLE Raw counts",las = 2)

#rawMedians <- if_else(rowMins(counts(analysis$dds))>0,rowMedians(counts(analysis$dds)),NA_real_)
#rawMedians <- rowMedians(counts(analysis$dds)[rowMins(counts(analysis$dds))>0,])
#rawLogRatios <- log(counts(analysis$dds)[rowMins(counts(analysis$dds))>0,]) - log(rawMedians)
rawMedianLogs <- rowMedians(log(counts(analysis$dds)[rowMins(counts(analysis$dds))>0,]))
rawLogRatios <- log(counts(analysis$dds)[rowMins(counts(analysis$dds))>0,]) - rawMedianLogs

as_tibble(rawLogRatios, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) +
  geom_boxplot(alpha = 0, outlier.shape = NA) +
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.25,2.33), expand = c(0,0)) +
  ggtitle("RLE Raw")

# plotRLE(counts(analysis$dds[rowMins(counts(analysis$dds))>0,], normalized = TRUE), outline = FALSE, main = "RLE Norm counts", las = 2)

#normMedians <- rowMedians(counts(analysis$dds, normalized = TRUE)[rowMins(counts(analysis$dds))>0,])
#normLogRatios <- log(counts(analysis$dds, normalized = TRUE)[rowMins(counts(analysis$dds))>0,]) - log(normMedians)
normMedianLogs <- rowMedians(log(counts(analysis$dds, normalized = TRUE)[rowMins(counts(analysis$dds))>0,]))
normLogRatios <- log(counts(analysis$dds, normalized = TRUE)[rowMins(counts(analysis$dds))>0,]) - normMedianLogs

as_tibble(normLogRatios, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) + 
  geom_boxplot(alpha = 0, outlier.shape = NA) + 
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.2,2.15), expand = c(0,0)) +
  ggtitle("RLE Normalized")

# Phase II

rawMedianLogs2 <- rowMedians(log(counts(analysis$dds2)[rowMins(counts(analysis$dds2))>0,]))
rawLogRatios2 <- log(counts(analysis$dds2)[rowMins(counts(analysis$dds2))>0,]) - rawMedianLogs2

as_tibble(rawLogRatios2, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) +
  geom_boxplot(alpha = 0, outlier.shape = NA) +
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.25,2.33), expand = c(0,0)) +
  ggtitle("Phase II RLE Raw")

normMedianLogs2 <- rowMedians(log(counts(analysis$dds2, normalized = TRUE)[rowMins(counts(analysis$dds2))>0,]))
normLogRatios2 <- log(counts(analysis$dds2, normalized = TRUE)[rowMins(counts(analysis$dds2))>0,]) - normMedianLogs2

as_tibble(normLogRatios2, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) + 
  geom_boxplot(alpha = 0, outlier.shape = NA) + 
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.2,2.15), expand = c(0,0)) +
  ggtitle("Phase II RLE Normalized")


normMedianLogs2alt <- rowMedians(log(counts(analysis$dds2alt, normalized = TRUE)[rowMins(counts(analysis$dds2alt))>0,]))
normLogRatios2alt <- log(counts(analysis$dds2alt, normalized = TRUE)[rowMins(counts(analysis$dds2alt))>0,]) - normMedianLogs2alt

as_tibble(normLogRatios2alt, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) + 
  geom_boxplot(alpha = 0, outlier.shape = NA) + 
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.2,2.15), expand = c(0,0)) +
  ggtitle("Phase II Alternate order RLE Normalized")


```


### Phase III? Uninfected Treated RLE

```{r}

rawMedianLogsUT <- rowMedians(log(counts(analysis$ddsUT)[rowMins(counts(analysis$ddsUT))>0,]))
rawLogRatiosUT <- log(counts(analysis$ddsUT)[rowMins(counts(analysis$ddsUT))>0,]) - rawMedianLogsUT

as_tibble(rawLogRatiosUT, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) +
  geom_boxplot(alpha = 0, outlier.shape = NA) +
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.25,2.33), expand = c(0,0)) +
  ggtitle("Phase II RLE Raw")

normMedianLogsUT <- rowMedians(log(counts(analysis$ddsUT, normalized = TRUE)[rowMins(counts(analysis$ddsUT))>0,]))
normLogRatiosUT <- log(counts(analysis$ddsUT, normalized = TRUE)[rowMins(counts(analysis$ddsUT))>0,]) - normMedianLogsUT

as_tibble(normLogRatiosUT, rownames = NA) %>% pivot_longer(everything(), names_to = "Sample", values_to = "RLE") %>% 
  ggplot(aes(x=Sample,y=RLE)) +
  geom_hline(yintercept = 0, color = "red") +
  geom_violin(draw_quantiles = c(0.25,0.75), trim = TRUE, color = "lightgreen", alpha = 0.1) + 
  geom_boxplot(alpha = 0, outlier.shape = NA) + 
  theme_bw() +
  theme(axis.text.x = element_text(vjust = 0.5,angle = 90, hjust = 1, size = 8, face = "bold"),
        axis.title.x = element_blank(),aspect.ratio = 0.55) +
  scale_y_continuous(limits = c(-2.2,2.15), expand = c(0,0)) +
  ggtitle("Phase II RLE Normalized")

```


# PCA

```{r pca, fig.height=11, fig.width=15}
analysis$pca <- pcaPlotGKT(analysis$vst[,analysis$vst$Source == "PBMC"],
                                  intgroup = c("SampleID",config$sidMetadata,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," PBMC PCA"))

analysis$pca

analysis$pca <- pcaPlotGKT(analysis$vst[,analysis$vst$Source == "BAL"],
                                  intgroup = c("SampleID",config$sidMetadata,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," BAL PCA"))

analysis$pca

# Phase II

analysis$pca2 <- pcaPlotGKT(analysis$vst2alt[,analysis$vst2alt$Source == "PBMC"],
                                  intgroup = c("SampleID",config$sidMetadata2,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II PBMC PCA"))

analysis$pca2

analysis$pca2 <- pcaPlotGKT(analysis$vst2alt[,analysis$vst2alt$Source == "BAL"],
                                  intgroup = c("SampleID",config$sidMetadata2,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II BAL PCA"))

analysis$pca2

phaseIPCA <- pcaPlotGKT(analysis$vst,
                                  intgroup = c("SampleID",config$sidMetadata,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = interaction(SubjectID, Source))) +
  geom_point(aes(shape = Source, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID), color = Treatment),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(22,23,24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase I PCA"))

ggsave(phaseIPCA, filename = "PhaseI_PCA.pdf", units = "in", width = 11, height = 8.5, scale = 1)

phaseIIPCA <- pcaPlotGKT(analysis$vst2,
                                  intgroup = c("SampleID",config$sidMetadata2,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = interaction(SubjectID, Source))) +
  geom_point(aes(shape = Source, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID), color = Treatment),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(22,23,24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II as labeled PCA"))

ggsave(phaseIIPCA, filename = "PhaseII_asLabeled_PCA.pdf", units = "in", width = 11, height = 8.5, scale = 1)

phaseIIPCAalt <- pcaPlotGKT(analysis$vst2alt,
                                  intgroup = c("SampleID",config$sidMetadata2,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = interaction(SubjectID, Source))) +
  geom_point(aes(shape = Source, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID), color = Treatment),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(22,23,24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II column reversed PCA"))

ggsave(phaseIIPCAalt, filename = "PhaseII_ColumnReversed_PCA.pdf", units = "in", width = 11, height = 8.5, scale = 1)


pcaPlotGKT(analysis$vstp12,
                                  intgroup = c("SampleID",config$sidMetadata,c("Day","Treatment")),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = interaction(SubjectID, Source))) +
  geom_point(aes(shape = Source, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8","lightgrey","grey","darkgrey")) + 
  geom_text_repel(aes(label = paste0(SubjectID), color = Treatment),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(22,23,24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis,"Both Phases column reversed PCA"))



```


### Phase III? Uninfected Treated PCA

```{r}

analysis$pcaUT <- pcaPlotGKT(analysis$vstUT[,analysis$vstUT$Source == "PBMC"],
                                  intgroup = c("SampleID",config$sidMetadataUT,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II PBMC PCA"))

analysis$pcaUT

analysis$pcaUT <- pcaPlotGKT(analysis$vstUT[,analysis$vstUT$Source == "BAL"],
                                  intgroup = c("SampleID",config$sidMetadataUT,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = SubjectID)) +
  geom_point(aes(shape = Treatment, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID)),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II BAL PCA"))

analysis$pcaUT


phaseIIPCAUT <- pcaPlotGKT(analysis$vstUT,
                                  intgroup = c("SampleID",config$sidMetadataUT,"Treatment"),
                                  xpc = 1, ypc = 2) +
  geom_path(aes(color = Treatment, group = interaction(SubjectID, Source))) +
  geom_point(aes(shape = Source, fill = Timepoint), size = 5) +
  scale_fill_manual(values = c("#53B400","#C49A00","#F8766D","#A58AFF","#FB61D7","#00C094","#00B6EB","#377eb8")) + 
  geom_text_repel(aes(label = paste0(SubjectID), color = Treatment),size = 4, hjust = 1, vjust = -1) +
  scale_shape_manual(values = c(22,23,24,25)) +
  scale_color_manual(values = c("green","red")) +
  scale_x_continuous(expand = c(0.5,0)) +
  guides(fill = guide_legend(order = 1, override.aes = list(shape = 21, size = 4))) +
  theme_bw() + ggtitle(paste0(analysis$config$analysis," Phase II as labeled PCA"))

phaseIIPCAUT
#ggsave(phaseIIPCA, filename = "PhaseII_asLabeled_PCA.pdf", units = "in", width = 11, height = 8.5, scale = 1)

```


## PBMC Analysis

```{r}

geneInfoTable <- read_tsv("/yerkes-cifs/runs/Genome_references/composite/macaca_mulatta_sars_cov2/Mmul10_MT246667.1/STAR2.7.3a/geneInfo.tab", col_names = c("gene_id","gene_symbol","gene_type"), skip = 1)
geneInfoTable <- column_to_rownames(geneInfoTable, "gene_id")
dupVec <- unlist(lapply(rownames(geneInfoTable), function(x) length(which(geneInfoTable$gene_symbol == geneInfoTable[x,]$gene_symbol))))
names(dupVec) <- rownames(geneInfoTable)
geneInfoTable$gene_symbol_uniq  = unlist(lapply(rownames(geneInfoTable), function(x) if_else(dupVec[x]>1,paste(geneInfoTable[x,]$gene_symbol,x, sep = "_"),geneInfoTable[x,]$gene_symbol)))
geneInfoTable <- rownames_to_column(geneInfoTable, "gene_id")

analysis$ddsPBMC <- DESeqDataSetFromMatrix(counts(analysis$dds[,analysis$dds$Source == "PBMC"]), colData(analysis$dds[,analysis$dds$Source == "PBMC"]), design = ~ SubjectInd + Treatment * Timepoint)
## This should work but sometimes sets rownames to NULL
rownames(analysis$ddsPBMC) <- geneInfoTable$gene_symbol_uniq
## This is an alternative
#names(analysis$ddsPBMC@rowRanges) <- geneInfoTable$gene_symbol_uniq

analysis$ddsPBMC <- DESeq(analysis$ddsPBMC, parallel = TRUE)
analysis$ddsPBMCresultsNames <- resultsNames(analysis$ddsPBMC)
analysis$ddsPBMCresultsNames

analysis$vstPBMC <-varianceStabilizingTransformation(analysis$ddsPBMC, blind = FALSE, fitType = "parametric")

#analysis$rldPBMC <- rlog(analysis$ddsPBMC, blind = FALSE, fitType = "parametric")

analysis$TimepointTreatmentMeansPBMC <- sapply(levels(interaction(analysis$ddsPBMC$Timepoint,
                                                                  analysis$ddsPBMC$Treatment)),
                                               function(lvl) rowMeans2(counts(analysis$ddsPBMC,
                                                                              normalize = TRUE)[,interaction(analysis$ddsPBMC$Timepoint,analysis$ddsPBMC$Treatment) == lvl]))
rownames(analysis$TimepointTreatmentMeansPBMC) <- rownames(analysis$ddsPBMC)
as.data.frame(analysis$TimepointTreatmentMeansPBMC) %>% rownames_to_column("gene") %>% pivot_longer(-gene,names_to = "TimepointTreatment",values_to = "mean_normalized_count") %>% mutate(Timepoint = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",1)), Treatment = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",2))) %>% str()

#assayPBMCforGSEA <- assay(analysis$rldPBMC)
#rownames(assayPBMCforGSEA)
#assayPBMCforGSEA <- assayPBMCforGSEA[rowMins(assayPBMCforGSEA)>0,]
#write_tsv(data.frame(Name = rownames(assayPBMCforGSEA),
#                     Description = "na",
#                     assayPBMCforGSEA),
#          path = "PBMC_rlog_forGSEA_COVten.txt")
#interaction_Treatment_Timepoint_PBMC <- #interaction(analysis$rldPBMC$Treatment,analysis$rldPBMC$Timepoint,sep = "_")
#clsLinesPBMC <- c(paste0(c(length(interaction_Treatment_Timepoint_PBMC),
#                          length(unique(interaction_Treatment_Timepoint_PBMC)),
#                          1), collapse = " "),
#                 paste0(c("#", unique(as.vector(interaction_Treatment_Timepoint_PBMC))), collapse = " "),
#                 paste0(interaction_Treatment_Timepoint_PBMC, collapse = " "))
#write_lines(clsLinesPBMC, path = "PBMC_Treatment_Timepoint_COVten.cls")

# Phase II

analysis$ddsPBMC2 <- DESeqDataSetFromMatrix(counts(analysis$dds2alt[,analysis$dds2alt$Source == "PBMC"]), colData(analysis$dds2alt[,analysis$dds2alt$Source == "PBMC"]), design = ~ SubjectInd + Treatment * Timepoint)
rownames(analysis$ddsPBMC2) <- geneInfoTable$gene_symbol_uniq
analysis$ddsPBMC2@rowRanges@partitioning@NAMES <- geneInfoTable$gene_symbol_uniq
analysis$ddsPBMC2 <- DESeq(analysis$ddsPBMC2, parallel = TRUE)

# Phase I and II

analysis$ddsPBMCp12 <- DESeqDataSetFromMatrix(counts(analysis$ddsp12[,analysis$ddsp12$Source == "PBMC"]), colData(analysis$ddsp12[,analysis$ddsp12$Source == "PBMC"]), design = ~ SubjectInd + Treatment * Timepoint)
rownames(analysis$ddsPBMCp12) <- geneInfoTable$gene_symbol_uniq
#analysis$ddsPBMCp12@rowRanges@partitioning@NAMES <- geneInfoTable$gene_symbol_uniq
analysis$ddsPBMCp12 <- DESeq(analysis$ddsPBMCp12, parallel = TRUE)

analysis$rldPBMCp12 <- rlog(analysis$ddsPBMCp12, blind = FALSE, fitType = "parametric")

assayPBMCp12forGSEA <- assay(analysis$rldPBMCp12)
assayPBMCp12forGSEA <- assayPBMCp12forGSEA[rowMins(assayPBMCp12forGSEA)>0,]
write_tsv(data.frame(Name = rownames(assayPBMCp12forGSEA),
                     Description = "na",
                     assayPBMCp12forGSEA),
          path = "PBMCp12_rlog_forGSEA_COVten.txt")
interactionPBMCp12_Treatment_Timepoint <- interaction(analysis$rldPBMCp12$Treatment,analysis$rldPBMCp12$Timepoint,sep = "_")
clsLinesPBMCp12 <- c(paste0(c(length(interactionPBMCp12_Treatment_Timepoint),
                          length(unique(interactionPBMCp12_Treatment_Timepoint)),
                          1), collapse = " "),
                 paste0(c("#", unique(as.vector(interactionPBMCp12_Treatment_Timepoint))), collapse = " "),
                 paste0(interactionPBMCp12_Treatment_Timepoint, collapse = " "))
write_lines(clsLinesPBMCp12, path = "PBMCp12_Treatment_Timepoint_COVten.cls")

analysis$rlogPBMCp12_Timepoint_Means <-
  sapply(levels(analysis$rldPBMCp12$Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldPBMCp12)[,analysis$rldPBMCp12$Timepoint == lvl]))
rownames(analysis$rlogPBMCp12_Timepoint_Means) <- rownames(analysis$rldPBMCp12)

analysis$rlogPBMCp12_FromMeanDay0 <- assay(analysis$rldPBMCp12) - 
  analysis$rlogPBMCp12_Timepoint_Means[,"Day0"]

analysis$rlogPBMCp12_FromMean <- assay(analysis$rldPBMCp12) - 
  rowMeans2(assay(analysis$rldPBMCp12))

analysis$rlogPBMCp12_Treatment_Timepoint_Means <-
  sapply(levels(interactionPBMCp12_Treatment_Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldPBMCp12)[,interactionPBMCp12_Treatment_Timepoint == lvl]))
rownames(analysis$rlogPBMCp12_Treatment_Timepoint_Means) <- rownames(analysis$rldPBMCp12)


```

### Phase III? Uninfected Treated

```{r}

analysis$ddsPBMCUT <- DESeqDataSetFromMatrix(counts(analysis$ddsUT[,analysis$ddsUT$Source == "PBMC"]), colData(analysis$ddsUT[,analysis$ddsUT$Source == "PBMC"]), design = ~ SubjectInd + Timepoint)
rownames(analysis$ddsPBMCUT) <- geneInfoTable$gene_symbol_uniq
analysis$ddsPBMCUT@rowRanges@partitioning@NAMES <- geneInfoTable$gene_symbol_uniq
analysis$ddsPBMCUT <- DESeq(analysis$ddsPBMCUT, parallel = TRUE)

analysis$rldPBMCUT <- rlog(analysis$ddsPBMCUT, blind = FALSE, fitType = "parametric")

assayPBMCUTforGSEA <- assay(analysis$rldPBMCUT)
assayPBMCUTforGSEA <- assayPBMCUTforGSEA[rowMins(assayPBMCUTforGSEA)>0,]
write_tsv(data.frame(Name = rownames(assayPBMCUTforGSEA),
                     Description = "na",
                     assayPBMCUTforGSEA),
          path = "PBMC_Uninf_Treat_rlog_forGSEA_COVten.txt")
interaction_Treatment_Timepoint_PBMCUT <- interaction(analysis$rldPBMCUT$Treatment,analysis$rldPBMCUT$Timepoint,sep = "_")
clsLinesPBMCUT <- c(paste0(c(length(interaction_Treatment_Timepoint_PBMCUT),
                          length(unique(interaction_Treatment_Timepoint_PBMCUT)),
                          1), collapse = " "),
                 paste0(c("#", unique(as.vector(interaction_Treatment_Timepoint_PBMCUT))), collapse = " "),
                 paste0(interaction_Treatment_Timepoint_PBMCUT, collapse = " "))
write_lines(clsLinesPBMCUT, path = "PBMC_Uninf_Treat_Treatment_Timepoint_COVten.cls")



analysis$rlogPBMCUT_Timepoint_Means <-
  sapply(levels(analysis$rldPBMCUT$Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldPBMCUT)[,analysis$rldPBMCUT$Timepoint == lvl]))
rownames(analysis$rlogPBMCUT_Timepoint_Means) <- rownames(analysis$rldPBMCUT)

analysis$rlogPBMCUT_FromMeanBL <- assay(analysis$rldPBMCUT) - 
  analysis$rlogPBMCUT_Timepoint_Means[,"BL"]

analysis$rlogPBMCUT_FromMeanDay0 <- assay(analysis$rldPBMCUT) - 
  analysis$rlogPBMCUT_Timepoint_Means[,"Day0"]

analysis$rlogPBMCUT_FromMean <- assay(analysis$rldPBMCUT) - 
  rowMeans2(assay(analysis$rldPBMCUT))


```


## PBMC plots

```{r}

linePlotsGKT(analysis$ddsPBMCp12, "IFI6", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
linePlotsGKT(analysis$ddsPBMCp12, "IFI27", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
linePlotsGKT(analysis$ddsPBMCp12, "ISG15", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()

# IDO1PBMC_lineplot <- linePlotsGKT(analysis$ddsPBMC, "IDO1", "SubjectID", "Timepoint", "Treatment", main = "IDO1 in PBMC") + scale_y_log10()
# ggsave(IDO1PBMC_lineplot, filename = "IDO1_inPBMC_lineplot.pdf", units = "in", width = 11, height = 8.5, scale = 1)
# 
# lineplotsPBMC <- list()
# plotgene = "TNFSF10"
# plotgene = "DSE"
# plotgene = "IRF1"
 plotgene = "IFNL1"
 lineplotsPBMC[[plotgene]] <- linePlotsGKT(analysis$ddsPBMC, plotgene, "SubjectID", "Timepoint", "Treatment", main = paste0(plotgene," in BAL")) + scale_y_log10()
 ggsave(lineplotsPBMC[[plotgene]], filename = paste0(plotgene,"_inPBMC_lineplot.pdf"), units = "in", width = 11, height = 8.5, scale = 1)


cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene == "ISG15") %>% ggplot(aes(x = Timepoint, y=lFCfromBL)) + geom_boxplot(aes(group = interaction(Timepoint,Treatment), color = Treatment), outlier.shape = NA) + geom_sina( aes(group = interaction(Timepoint,Treatment), color = Treatment)) + 
#  geom_text_repel(aes(group = interaction(Timepoint,Treatment),label = SubjectID), size = 3) +
  theme_bw()



```



## PBMC Results

```{r}

tmpRes <- results(analysis$ddsPBMC, contrast = list("Timepoint_Day0_vs_BL"), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day0","BL")))
tmpRes2 <- results(analysis$ddsPBMC, contrast = list("Timepoint_Day2_vs_BL","Timepoint_Day0_vs_BL"), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day2","Day0")))

summary(results(analysis$ddsPBMC, contrast = list( c("Treatment_IFNant_vs_Untreated")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Treatment", comp = c("IFNant","Untreated"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Treatment_IFNant_vs_Untreated", "TreatmentIFNant.TimepointDay0")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Treatment", comp = c("IFNant","Untreated"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Treatment_IFNant_vs_Untreated", "TreatmentIFNant.TimepointDay1")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Treatment", comp = c("IFNant","Untreated"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Treatment_IFNant_vs_Untreated", "TreatmentIFNant.TimepointDay2")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Treatment", comp = c("IFNant","Untreated"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Treatment_IFNant_vs_Untreated", "TreatmentIFNant.TimepointDay4")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Treatment", comp = c("IFNant","Untreated"))))



summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day0_vs_BL")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day0","BL"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day0_vs_BL", "TreatmentIFNant.TimepointDay0")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day0","BL"))))

summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day1_vs_BL")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day1","BL"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day1_vs_BL", "TreatmentIFNant.TimepointDay1")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day1","BL"))))

summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day2_vs_BL")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day2","BL"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day2_vs_BL", "TreatmentIFNant.TimepointDay2")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day2","BL"))))

summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day4_vs_BL")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day4","BL"))))
summary(results(analysis$ddsPBMC, contrast = list( c("Timepoint_Day4_vs_BL", "TreatmentIFNant.TimepointDay4")), alpha = 0.05, filter = maxMinFilter(analysis$ddsPBMC, intgroup = "Timepoint", comp = c("Day4","BL"))))

res_PBMC_IFNant_Day0_vs_DayM7 <- results(analysis$ddsPBMC,
                                         contrast = list( c("Timepoint_Day0_vs_BL",
                                                            "TreatmentIFNant.TimepointDay0")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMC,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))

res_PBMC_Untreated_Day0_vs_DayM7 <- results(analysis$ddsPBMC,
                                         contrast = list( c("Timepoint_Day0_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMC,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))

res_PBMC_Treatment_Interaction_Day0 <- results(analysis$ddsPBMC,
                                               contrast = list( c("TreatmentIFNant.TimepointDay0")),
                                               alpha = 0.05,
                                               filter = maxMinFilter(analysis$ddsPBMC,
                                                                     intgroup = "Timepoint",
                                                                     comp = c("Day0","BL")))
res_PBMC_DayM7_IFNant_vs_Untreated <- results(analysis$ddsPBMC,
                                              contrast = list( c("Treatment_IFNant_vs_Untreated")),
                                              alpha = 0.05,
                                              filter = maxMinFilter(analysis$ddsPBMC,
                                                                    intgroup = "Treatment",
                                                                    comp = c("IFNant","Untreated")))

res_PBMC_Day0_IFNant_vs_Untreated <- results(analysis$ddsPBMC,
                                              contrast = list( c("Treatment_IFNant_vs_Untreated",
                                                                 "TreatmentIFNant.TimepointDay0")),
                                              alpha = 0.05,
                                              filter = maxMinFilter(analysis$ddsPBMC,
                                                                    intgroup = "Treatment",
                                                                    comp = c("IFNant","Untreated")))

merged_PBMC_D0_results <- as.annot.data.frame(res_PBMC_IFNant_Day0_vs_DayM7,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day0_vs_BL+TreatmentIFNant.TimepointDay0 effect",
                                              newlabel = "IFNant  Day0/BL") %>%
  left_join(as.annot.data.frame(res_PBMC_Untreated_Day0_vs_DayM7,
                                subset = "PBMC",
                                relabel = "Timepoint_Day0_vs_BL effect",
                                newlabel = "Untreated Day0/BL")) %>%
  left_join(as.annot.data.frame(res_PBMC_Day0_IFNant_vs_Untreated,
                                subset = "PBMC",
                                relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay0 effect",
                                newlabel = "Day0 IFNant/Untreated")) %>%
  left_join(as.annot.data.frame(res_PBMC_DayM7_IFNant_vs_Untreated,
                                subset = "PBMC",
                                relabel = "Treatment_IFNant_vs_Untreated effect",
                                newlabel = "BL IFNant/Untreated")) %>%
  left_join(as.annot.data.frame(res_PBMC_Treatment_Interaction_Day0,
                                subset = "PBMC",
                                relabel = "TreatmentIFNant.TimepointDay0 effect",
                                newlabel = "Treatment Timepoint Interaction"))
  arrange(merged_PBMC_D0_results, `padj (BH adjusted p-values: IFNant  Day0/BL)`) %>% 
    write.xlsx(file = "IFNant_Day0_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  
## Phase I and II  

resultsNames(analysis$ddsPBMCp12)

# Day0 PBMC

res_PBMCp12_Untreated_Day0_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day0_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))
res_PBMCp12_IFNant_Day0_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day0_vs_BL",
                                                            "TreatmentIFNant.TimepointDay0")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))

res_PBMCp12_Day0_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay0")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))

res_PBMCp12_Day0_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay0")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day0","BL")))

merged_PBMCp12_D0_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day0_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day0_vs_BL+TreatmentIFNant.TimepointDay0 effect",
                                              newlabel = "IFNant Day0/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day0_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day0_vs_BL effect",
                                              newlabel = "Untreated Day0/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day0_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay0 effect",
                                              newlabel = "Day0 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day0_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay0 effect",
                                              newlabel = "Day0 IFNant/Untreated"))

arrange(merged_PBMCp12_D0_results, `padj (BH adjusted p-values: Day0 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day0_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  
## preranked GSEA input files
write_tsv(as.data.frame(res_PBMCp12_Day0_Treatment_Interaction[res_PBMCp12_Day0_Treatment_Interaction$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_Day0_Treatment_Interaction.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_PBMCp12_Untreated_Day0_vs_BL[res_PBMCp12_Untreated_Day0_vs_BL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_Untreated_Day0_vs_BL.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_PBMCp12_IFNant_Day0_vs_DayBL[res_PBMCp12_IFNant_Day0_vs_DayBL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_IFNant_Day0_vs_BL.rnk", col_names = FALSE)



# Day1 PBMC

res_PBMCp12_Untreated_Day1_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day1_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day1","BL")))
res_PBMCp12_IFNant_Day1_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day1_vs_BL",
                                                            "TreatmentIFNant.TimepointDay1")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day1","BL")))

res_PBMCp12_Day1_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay1")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day1","BL")))

res_PBMCp12_Day1_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay1")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day1","BL")))

merged_PBMCp12_D1_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day1_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day1_vs_BL+TreatmentIFNant.TimepointDay1 effect",
                                              newlabel = "IFNant Day1/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day1_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day1_vs_BL effect",
                                              newlabel = "Untreated Day1/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day1_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay1 effect",
                                              newlabel = "Day1 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day1_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay1 effect",
                                              newlabel = "Day1 IFNant/Untreated"))

arrange(merged_PBMCp12_D1_results, `padj (BH adjusted p-values: Day1 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day1_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  

# Day2 PBMC

res_PBMCp12_Untreated_Day2_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day2_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))
res_PBMCp12_IFNant_Day2_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day2_vs_BL",
                                                            "TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

res_PBMCp12_Day2_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

res_PBMCp12_Day2_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

merged_PBMCp12_D2_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day2_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day2_vs_BL+TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "IFNant Day2/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day2_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day2_vs_BL effect",
                                              newlabel = "Untreated Day2/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day2_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "Day2 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day2_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "Day2 IFNant/Untreated"))

arrange(merged_PBMCp12_D2_results, `padj (BH adjusted p-values: Day2 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day2_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  
## preranked GSEA input files
write_tsv(as.data.frame(res_PBMCp12_Day2_Treatment_Interaction[res_PBMCp12_Day2_Treatment_Interaction$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_Day2_Treatment_Interaction.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_PBMCp12_Untreated_Day2_vs_BL[res_PBMCp12_Untreated_Day2_vs_BL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_Untreated_Day2_vs_BL.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_PBMCp12_IFNant_Day2_vs_DayBL[res_PBMCp12_IFNant_Day2_vs_DayBL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_PBMCp12_IFNant_Day2_vs_BL.rnk", col_names = FALSE)



# Day4 PBMC

res_PBMCp12_Untreated_Day4_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day4_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))
res_PBMCp12_IFNant_Day4_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day4_vs_BL",
                                                            "TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

res_PBMCp12_Day4_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

res_PBMCp12_Day4_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

merged_PBMCp12_D4_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day4_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day4_vs_BL+TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "IFNant Day4/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day4_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day4_vs_BL effect",
                                              newlabel = "Untreated Day4/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day4_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "Day4 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day4_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "Day4 IFNant/Untreated"))

arrange(merged_PBMCp12_D4_results, `padj (BH adjusted p-values: Day4 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day4_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  



# Day5 PBMC

res_PBMCp12_Untreated_Day5_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day5_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day5","BL")))
res_PBMCp12_IFNant_Day5_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day5_vs_BL",
                                                            "TreatmentIFNant.TimepointDay5")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day5","BL")))

res_PBMCp12_Day5_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay5")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day5","BL")))

res_PBMCp12_Day5_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay5")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day5","BL")))

merged_PBMCp12_D5_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day5_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day5_vs_BL+TreatmentIFNant.TimepointDay5 effect",
                                              newlabel = "IFNant Day5/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day5_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day5_vs_BL effect",
                                              newlabel = "Untreated Day5/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day5_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay5 effect",
                                              newlabel = "Day5 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day5_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay5 effect",
                                              newlabel = "Day5 IFNant/Untreated"))

arrange(merged_PBMCp12_D5_results, `padj (BH adjusted p-values: Day5 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day5_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  



# Day7 PBMC

res_PBMCp12_Untreated_Day7_vs_BL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day7_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))
res_PBMCp12_IFNant_Day7_vs_DayBL <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Timepoint_Day7_vs_BL",
                                                            "TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

res_PBMCp12_Day7_Treatment_Interaction <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

res_PBMCp12_Day7_IFNantVsUntreated <- results(analysis$ddsPBMCp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsPBMCp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

merged_PBMCp12_D7_results <- as.annot.data.frame(res_PBMCp12_IFNant_Day7_vs_DayBL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day7_vs_BL+TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "IFNant Day7/BL") %>%
  left_join(as.annot.data.frame(res_PBMCp12_Untreated_Day7_vs_BL,
                                              subset = "PBMC",
                                              relabel = "Timepoint_Day7_vs_BL effect",
                                              newlabel = "Untreated Day7/BL")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day7_Treatment_Interaction,
                                              subset = "PBMC",
                                              relabel = "TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "Day7 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_PBMCp12_Day7_IFNantVsUntreated,
                                              subset = "PBMC",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "Day7 IFNant/Untreated"))

arrange(merged_PBMCp12_D7_results, `padj (BH adjusted p-values: Day7 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day7_PBMC_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  


```

## BAL Analysis

```{r}

analysis$ddsBAL <- DESeqDataSetFromMatrix(counts(analysis$dds[,analysis$dds$Source == "BAL"]), colData(analysis$dds[,analysis$dds$Source == "BAL"]), design = ~ SubjectInd + Treatment * Timepoint)
rownames(analysis$ddsBAL) <- geneInfoTable$gene_symbol_uniq
analysis$ddsBAL <- DESeq(analysis$ddsBAL, parallel = TRUE)
analysis$ddsBALresultsNames <- resultsNames(analysis$ddsBAL)
analysis$ddsBALresultsNames

analysis$vstBAL <-varianceStabilizingTransformation(analysis$ddsBAL, blind = FALSE, fitType = "parametric")

analysis$rldBAL <- rlog(analysis$ddsBAL, blind = FALSE, fitType = "parametric")

assayBALforGSEA <- assay(analysis$rldBAL)
assayBALforGSEA <- assayBALforGSEA[rowMins(assayBALforGSEA)>0,]
write_tsv(data.frame(Name = rownames(assayBALforGSEA),
                     Description = "na",
                     assayBALforGSEA),
          path = "BAL_rlog_forGSEA_COVten.txt")
interaction_Treatment_Timepoint <- interaction(analysis$rldBAL$Treatment,analysis$rldBAL$Timepoint,sep = "_")
clsLinesBAL <- c(paste0(c(length(interaction_Treatment_Timepoint),
                          length(unique(interaction_Treatment_Timepoint)),
                          1), collapse = " "),
                 paste0(c("#", unique(as.vector(interaction_Treatment_Timepoint))), collapse = " "),
                 paste0(interaction_Treatment_Timepoint, collapse = " "))
write_lines(clsLinesBAL, path = "BAL_Treatment_Timepoint_COVten.cls")

# Phase II

analysis$ddsBAL2 <- DESeqDataSetFromMatrix(counts(analysis$dds2alt[,analysis$dds2alt$Source == "BAL"]), colData(analysis$dds2alt[,analysis$dds2alt$Source == "BAL"]), design = ~ SubjectInd + Treatment * Timepoint)
rownames(analysis$ddsBAL2) <- geneInfoTable$gene_symbol_uniq
analysis$ddsBAL2 <- DESeq(analysis$ddsBAL2, parallel = TRUE)

# Phase I and II

analysis$ddsBALp12 <- DESeqDataSetFromMatrix(counts(analysis$ddsp12[,analysis$ddsp12$Source == "BAL"]), colData(analysis$ddsp12[,analysis$ddsp12$Source == "BAL"]), design = ~ SubjectInd + Treatment * Timepoint)
rownames(analysis$ddsBALp12) <- geneInfoTable$gene_symbol_uniq
analysis$ddsBALp12 <- DESeq(analysis$ddsBALp12, parallel = TRUE)

analysis$rldBALp12 <- rlog(analysis$ddsBALp12, blind = FALSE, fitType = "parametric")

## need to edit for BAL Phase I and II
assayBALp12forGSEA <- assay(analysis$rldBALp12)
assayBALp12forGSEA <- assayBALp12forGSEA[rowMins(assayBALp12forGSEA)>0,]
write_tsv(data.frame(Name = rownames(assayBALp12forGSEA),
                     Description = "na",
                     assayBALp12forGSEA),
          path = "BALp12_rlog_forGSEA_COVten.txt")
interactionBALp12_Treatment_Timepoint <- interaction(analysis$rldBALp12$Treatment,analysis$rldBALp12$Timepoint,sep = "_")
clsLinesBALp12 <- c(paste0(c(length(interactionBALp12_Treatment_Timepoint),
                          length(unique(interactionBALp12_Treatment_Timepoint)),
                          1), collapse = " "),
                 paste0(c("#", unique(as.vector(interactionBALp12_Treatment_Timepoint))), collapse = " "),
                 paste0(interactionBALp12_Treatment_Timepoint, collapse = " "))
write_lines(clsLinesBALp12, path = "BALp12_Treatment_Timepoint_COVten.cls")



analysis$rlogBALp12_Timepoint_Means <-
  sapply(levels(analysis$rldBALp12$Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldBALp12)[,analysis$rldBALp12$Timepoint == lvl]))
rownames(analysis$rlogBALp12_Timepoint_Means) <- rownames(analysis$rldBALp12)

analysis$rlogBALp12_FromMean <- assay(analysis$rldBALp12) - 
  rowMeans2(assay(analysis$rldBALp12))

analysis$rlogBALp12_Treatment_Timepoint_Means <-
  sapply(levels(interactionBALp12_Treatment_Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldBALp12)[,interactionBALp12_Treatment_Timepoint == lvl]))
rownames(analysis$rlogBALp12_Treatment_Timepoint_Means) <- rownames(analysis$rldBALp12)




cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% filter(gene == "ISG15") %>% str()

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IFI6","ISG15")) %>% ggplot(aes(x = Treatment, y=lFCfromBL)) + geom_boxplot(aes(group = interaction(Timepoint,Treatment), color = Treatment), outlier.shape = NA) + geom_jitter( aes(group = interaction(Timepoint,Treatment), color = Treatment)) + 
#  geom_text_repel(position = "jitter",aes(group = interaction(Timepoint,Treatment),label = SubjectID), size = 3) +
  theme_bw() +
#  facet_col( ~ gene) +
  facet_wrap( ~ gene * Timepoint)

facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggboxplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", add = "jitter", outlier.shape = NA), facet.by = c("gene")) + theme_bw()

```

### Phase III? Uninfected Treated

```{r}

analysis$ddsBALUT <- DESeqDataSetFromMatrix(counts(analysis$ddsUT[,analysis$ddsUT$Source == "BAL"]), colData(analysis$ddsUT[,analysis$ddsUT$Source == "BAL"]), design = ~ SubjectInd + Timepoint)
rownames(analysis$ddsBALUT) <- geneInfoTable$gene_symbol_uniq
#analysis$ddsBALUT@rowRanges@partitioning@NAMES <- geneInfoTable$gene_symbol_uniq
analysis$ddsBALUT <- DESeq(analysis$ddsBALUT, parallel = TRUE)

analysis$rldBALUT <- rlog(analysis$ddsBALUT, blind = FALSE, fitType = "parametric")

assayBALUTforGSEA <- assay(analysis$rldBALUT)
assayBALUTforGSEA <- assayBALUTforGSEA[rowMins(assayBALUTforGSEA)>0,]
write_tsv(data.frame(Name = rownames(assayBALUTforGSEA),
                     Description = "na",
                     assayBALUTforGSEA),
          path = "BAL_Uninf_Treat_rlog_forGSEA_COVten.txt")
interaction_Treatment_Timepoint_BALUT <- interaction(analysis$rldBALUT$Treatment,analysis$rldBALUT$Timepoint,sep = "_")
clsLinesBALUT <- c(paste0(c(length(interaction_Treatment_Timepoint_BALUT),
                          length(unique(interaction_Treatment_Timepoint_BALUT)),
                          1), collapse = " "),
                 paste0(c("#", unique(as.vector(interaction_Treatment_Timepoint_BALUT))), collapse = " "),
                 paste0(interaction_Treatment_Timepoint_BALUT, collapse = " "))
write_lines(clsLinesBALUT, path = "BAL_Uninf_Treat_Treatment_Timepoint_COVten.cls")



analysis$rlogBALUT_Timepoint_Means <-
  sapply(levels(analysis$rldBALUT$Timepoint),
         function(lvl) rowMeans2(assay(analysis$rldBALUT)[,analysis$rldBALUT$Timepoint == lvl]))
rownames(analysis$rlogBALUT_Timepoint_Means) <- rownames(analysis$rldBALUT)

analysis$rlogBALUT_FromMeanBL <- assay(analysis$rldBALUT) - 
  analysis$rlogBALUT_Timepoint_Means[,"BL"]

analysis$rlogBALUT_FromMeanDay0 <- assay(analysis$rldBALUT) - 
  analysis$rlogBALUT_Timepoint_Means[,"Day0"]

analysis$rlogBALUT_FromMean <- assay(analysis$rldBALUT) - 
  rowMeans2(assay(analysis$rldBALUT))


```


## BAL Results

```{r}

resultsNames(analysis$ddsBALp12)

# Day2 BAL

res_BALp12_Untreated_Day2_vs_BL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day2_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))
res_BALp12_IFNant_Day2_vs_DayBL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day2_vs_BL",
                                                            "TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

res_BALp12_Day2_Treatment_Interaction <- results(analysis$ddsBALp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

res_BALp12_Day2_IFNantVsUntreated <- results(analysis$ddsBALp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay2")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day2","BL")))

merged_BALp12_D2_results <- as.annot.data.frame(res_BALp12_IFNant_Day2_vs_DayBL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day2_vs_BL+TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "IFNant Day2/BL") %>%
  left_join(as.annot.data.frame(res_BALp12_Untreated_Day2_vs_BL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day2_vs_BL effect",
                                              newlabel = "Untreated Day2/BL")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day2_Treatment_Interaction,
                                              subset = "BAL",
                                              relabel = "TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "Day2 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day2_IFNantVsUntreated,
                                              subset = "BAL",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay2 effect",
                                              newlabel = "Day2 IFNant/Untreated"))

arrange(merged_BALp12_D2_results, `padj (BH adjusted p-values: Day2 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day2_BAL_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  
## preranked GSEA input files
write_tsv(as.data.frame(res_BALp12_Day2_Treatment_Interaction[res_BALp12_Day2_Treatment_Interaction$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_BALp12_Day2_Treatment_Interaction.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_BALp12_Untreated_Day2_vs_BL[res_BALp12_Untreated_Day2_vs_BL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_BALp12_Untreated_Day2_vs_BL.rnk", col_names = FALSE)

write_tsv(as.data.frame(res_BALp12_IFNant_Day2_vs_DayBL[res_BALp12_IFNant_Day2_vs_DayBL$baseMean > 1,]) %>% arrange(desc(stat)) %>% rownames_to_column(var = "gene_symbol") %>% select(c("gene_symbol","stat")),path = "res_BALp12_IFNant_Day2_vs_BL.rnk", col_names = FALSE)



# Day4 BAL

res_BALp12_Untreated_Day4_vs_BL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day4_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))
res_BALp12_IFNant_Day4_vs_DayBL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day4_vs_BL",
                                                            "TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

res_BALp12_Day4_Treatment_Interaction <- results(analysis$ddsBALp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

res_BALp12_Day4_IFNantVsUntreated <- results(analysis$ddsBALp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay4")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day4","BL")))

merged_BALp12_D4_results <- as.annot.data.frame(res_BALp12_IFNant_Day4_vs_DayBL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day4_vs_BL+TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "IFNant Day4/BL") %>%
  left_join(as.annot.data.frame(res_BALp12_Untreated_Day4_vs_BL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day4_vs_BL effect",
                                              newlabel = "Untreated Day4/BL")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day4_Treatment_Interaction,
                                              subset = "BAL",
                                              relabel = "TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "Day4 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day4_IFNantVsUntreated,
                                              subset = "BAL",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay4 effect",
                                              newlabel = "Day4 IFNant/Untreated"))

arrange(merged_BALp12_D4_results, `padj (BH adjusted p-values: Day4 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day4_BAL_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  


# Day7 BAL

res_BALp12_Untreated_Day7_vs_BL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day7_vs_BL")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))
res_BALp12_IFNant_Day7_vs_DayBL <- results(analysis$ddsBALp12,
                                         contrast = list( c("Timepoint_Day7_vs_BL",
                                                            "TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

res_BALp12_Day7_Treatment_Interaction <- results(analysis$ddsBALp12,
                                         contrast = list( c("TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

res_BALp12_Day7_IFNantVsUntreated <- results(analysis$ddsBALp12,
                                         contrast = list( c("Treatment_IFNant_vs_Untreated",
                                           "TreatmentIFNant.TimepointDay7")),
                                         alpha = 0.05,
                                         filter = maxMinFilter(analysis$ddsBALp12,
                                                               intgroup = "Timepoint",
                                                               comp = c("Day7","BL")))

merged_BALp12_D7_results <- as.annot.data.frame(res_BALp12_IFNant_Day7_vs_DayBL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day7_vs_BL+TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "IFNant Day7/BL") %>%
  left_join(as.annot.data.frame(res_BALp12_Untreated_Day7_vs_BL,
                                              subset = "BAL",
                                              relabel = "Timepoint_Day7_vs_BL effect",
                                              newlabel = "Untreated Day7/BL")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day7_Treatment_Interaction,
                                              subset = "BAL",
                                              relabel = "TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "Day7 Treat-Time Interaction")) %>%
  left_join(as.annot.data.frame(res_BALp12_Day7_IFNantVsUntreated,
                                              subset = "BAL",
                                              relabel = "Treatment_IFNant_vs_Untreated+TreatmentIFNant.TimepointDay7 effect",
                                              newlabel = "Day7 IFNant/Untreated"))

arrange(merged_BALp12_D7_results, `padj (BH adjusted p-values: Day7 Treat-Time Interaction)`) %>% 
    write.xlsx(file = "Day7_BAL_P12_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)
  




```


## BAL Plots

```{r}

linePlotsGKT(analysis$ddsBALp12, "IFI6", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
linePlotsGKT(analysis$ddsBALp12, "IFI27", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
linePlotsGKT(analysis$ddsBALp12, "ISG15", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
linePlotsGKT(analysis$ddsBALp12, "TNFSF10", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()
# IDO1BAL_lineplot <- linePlotsGKT(analysis$ddsBAL, "IDO1", "SubjectID", "Timepoint", "Treatment", main = "IDO1 in BAL") + scale_y_log10()
# ggsave(IDO1BAL_lineplot, filename = "IDO1_inBAL_lineplot.pdf", units = "in", width = 11, height = 8.5, scale = 1)
# lineplots <- list()
# plotgene = "TNFSF10"
# plotgene = "DSE"
# plotgene = "IRF1"
plotgene = "IFNL1"
lineplots[[plotgene]] <- linePlotsGKT(analysis$ddsBALp12, plotgene, "SubjectID", "Timepoint", "Treatment", main = paste0(plotgene," in BAL")) + scale_y_log10()
# ggsave(lineplots[[plotgene]], filename = paste0(plotgene,"_inBAL_lineplot.pdf"), units = "in", width = 11, height = 8.5, scale = 1)


linePlotDat <- data.frame(count = t(counts(analysis$ddsBALp12, normalized = TRUE)[c("IFI6","IFI27","ISG15"),]), colData(analysis$ddsBALp12))
linePlotDat <- data.frame(count = counts(analysis$ddsBALp12, normalized = TRUE)["IFI6",], colData(analysis$ddsBALp12))

cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[c("IFI6","IFI27","ISG15"),])) %>% as.data.frame() %>% pivot_longer( c(13:15), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_grid(~ gene) + scale_y_log10()

cbind(colData(analysis$ddsBALp12),IFI6 = counts(analysis$ddsBALp12, normalized = TRUE)[c("IFI6"),]) %>% as.data.frame() %>% pivot_longer( c(13), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_grid(~ gene) + scale_y_log10()

```


## Levin Genesets

```{r fig.height=8.5, fig.width=11}

LevinList <- list()
LevinList$ISGs <- read_lines("LevinEtAl_ISG_List.txt",skip_empty_rows = TRUE)

setdiff(LevinList$ISGs, rownames(analysis$ddsPBMCp12))
LevinList$Tunable <- LevinList$ISGs[1:10]
LevinList$Robust <- LevinList$ISGs[11:26]
robust <- c("IRF7","IFI35","IFIH1","IRF9","EIF2AK2","OAS1","OAS2","DDX58","PLSCR1","IFIT3","IFIT2","MX1","IFIT1","IFI6","IFITM1","ISG15")
tunable <- c("IDO1","IDO2","CCR1","IL11","CXCL2","CCL20","IL6","IL8","CXCL11","CXCL10","HRASLS2","TNFSF13B","CEACAM1","IFI16","IRF1","TNFSF10","TNFAIP3")
enhancing <- c("IFITM1","IFITM2","IFITM3","NRP1","SIGLEC1","CD209","CLEC4M","ACE2","TMPRSS2","TMPRSS5","CTSA","CTSB","CTSC","CTSE","CTSF","CTSG","CTSH","CTSK","CTSL","CTSS","CTSV","CTSW","CTSZ")
inhibitory <- c("LY6E","GBPS","ZC3HAV1","BST2")

ISGList <- c("IFI6","IFI27","ISG15","MX1","ISG20","MX2","STAT1","GBP1","OAS2","IFIT3","IFIT2","IFI44L","IRF7","OAS1","OAS3","RSAD2","IFIT5","CXCL10")

cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(LevinList$ISGs,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(LevinList$ISGs,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene) + scale_y_log10() + theme_bw()

levintunableBALp12 <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(LevinList$Tunable,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(LevinList$Tunable,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Levin Tunable ISGs in BAL")
ggsave(levintunableBALp12, filename = "Levin_Tunable_ISGs_inBAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)
levintunableBALp12

levintunablePBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(LevinList$Tunable,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(LevinList$Tunable,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Levin Tunable ISGs in PBMC")
ggsave(levintunablePBMC, filename = "Levin_Tunable_ISGs_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)
levintunablePBMC

levinrobustBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(LevinList$Robust,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(LevinList$Robust,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Levin Robust ISGs in BAL")
ggsave(levinrobustBAL, filename = "Levin_Robust_ISGs_inBAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)
levinrobustBAL

levinrobustPBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(LevinList$Robust,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(LevinList$Robust,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Levin Robust ISGs in PBMC")
ggsave(levinrobustPBMC, filename = "Levin_Robust_ISGs_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)
levinrobustPBMC

amitgoi1 <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(c("IFNA13","IFNA8","IFNA6","IFNA14","IFNA10","IFNB1","IFNG","IFNK","IFNW1","IFNL1"),rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:22), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene) + scale_y_log10() + theme_bw()
ggsave(amitgoi1, filename = "amitgoi1_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)
amitgoi1

amitgoi1BAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(c("IFNA13","IFNA8","IFNA6","IFNA14","IFNA10","IFNB1","IFNG","IFNK","IFNW1","IFNL1"),rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:22), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene) + scale_y_log10() + theme_bw()
ggsave(amitgoi1BAL, filename = "amitgoi1_inBAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)
amitgoi1BAL

## plots for Nov Presentation

robustPBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(robust,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(robust,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I robust ISGs in PBMC")
ggsave(robustPBMC, filename = "PhaseI_robust_inPBMC_lineplot.pdf", units = "in", width = 11, height = 8.5, scale = 1)

robustPBMC2 <- cbind(colData(analysis$ddsPBMC2),t(counts(analysis$ddsPBMC2, normalized = TRUE)[intersect(robust,rownames(analysis$ddsPBMC2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(robust,rownames(analysis$ddsPBMC))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II robust ISGs in PBMC")
ggsave(robustPBMC2, filename = "PhaseII_robust_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

robustBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(robust,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(robust,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I robust ISGs in BAL")
ggsave(robustBAL, filename = "PhaseI_robust_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

robustBAL2 <- cbind(colData(analysis$ddsBAL2),t(counts(analysis$ddsBAL2, normalized = TRUE)[intersect(robust,rownames(analysis$ddsBAL2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(robust,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II robust ISGs in BAL")
ggsave(robustBAL2, filename = "PhaseII_robust_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

tunablePBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(tunable,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(tunable,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I tunable ISGs in PBMC")
ggsave(tunablePBMC, filename = "PhaseI_tunable_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

tunablePBMC2 <- cbind(colData(analysis$ddsPBMC2),t(counts(analysis$ddsPBMC2, normalized = TRUE)[intersect(tunable,rownames(analysis$ddsPBMC2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(tunable,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II tunable ISGs in PBMC")
tunablePBMC2 <- cbind(colData(analysis$ddsPBMC2),t(counts(analysis$ddsPBMC2, normalized = TRUE)[intersect(tunable,rownames(analysis$ddsPBMC2)),])) %>% as.data.frame() %>% pivot_longer( c(12:14,16:24), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II tunable ISGs in PBMC")
ggsave(tunablePBMC2, filename = "PhaseII_tunable_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

tunableBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(tunable,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(tunable,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I tunable ISGs in BAL")
ggsave(tunableBAL, filename = "PhaseI_tunable_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

tunableBAL2 <- cbind(colData(analysis$ddsBAL2),t(counts(analysis$ddsBAL2, normalized = TRUE)[intersect(tunable,rownames(analysis$ddsBAL2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(tunable,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II tunable ISGs in BAL")
ggsave(tunableBAL2, filename = "PhaseII_tunable_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

enhancingPBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(enhancing,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(enhancing,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I enhancing ISGs in PBMC")
ggsave(enhancingPBMC, filename = "PhaseI_enhancing_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

enhancingPBMC2 <- cbind(colData(analysis$ddsPBMC2),t(counts(analysis$ddsPBMC2, normalized = TRUE)[intersect(enhancing,rownames(analysis$ddsPBMC2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(enhancing,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II enhancing ISGs in PBMC")
ggsave(enhancingPBMC2, filename = "PhaseII_enhancing_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

enhancingBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(enhancing,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(enhancing,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I enhancing ISGs in BAL")
ggsave(enhancingBAL, filename = "PhaseI_enhancing_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

enhancingBAL2 <- cbind(colData(analysis$ddsBAL2),t(counts(analysis$ddsBAL2, normalized = TRUE)[intersect(enhancing,rownames(analysis$ddsBAL2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(enhancing,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II enhancing ISGs in BAL")
ggsave(enhancingBAL2, filename = "PhaseII_enhancing_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

inhibitoryPBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(inhibitory,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(inhibitory,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I inhibitory ISGs in PBMC")
ggsave(inhibitoryPBMC, filename = "PhaseI_inhibitory_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

inhibitoryPBMC2 <- cbind(colData(analysis$ddsPBMC2),t(counts(analysis$ddsPBMC2, normalized = TRUE)[intersect(inhibitory,rownames(analysis$ddsPBMC2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(inhibitory,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II inhibitory ISGs in PBMC")
ggsave(inhibitoryPBMC2, filename = "PhaseII_inhibitory_inPBMC_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

inhibitoryBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(inhibitory,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(inhibitory,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase I inhibitory ISGs in BAL")
ggsave(inhibitoryBAL, filename = "PhaseI_inhibitory_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

inhibitoryBAL2 <- cbind(colData(analysis$ddsBAL2),t(counts(analysis$ddsBAL2, normalized = TRUE)[intersect(inhibitory,rownames(analysis$ddsBAL2)),])) %>% as.data.frame() %>% pivot_longer( c(12:(11 + length(intersect(inhibitory,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + geom_text_repel(aes(label = SubjectID), size = 2) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "Phase II inhibitory ISGs in BAL")
ggsave(inhibitoryBAL2, filename = "PhaseII_inhibitory_inBAL_lineplot.pdf", units = "in", width = 14, height = 9, scale = 1)

###

linePlotsGKT(analysis$ddsPBMCp12, "IFNL1", "SubjectID", "Timepoint", "Treatment") + scale_y_log10()

ifngenes <- read_lines("IFNs_in_CATAnnotation.txt")
setdiff(ifngenes,rownames(analysis$ddsBALp12))
intersect(ifngenes,rownames(analysis$ddsBALp12))

ifnBAL <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(ifngenes,rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(ifngenes,rownames(analysis$ddsBALp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "IFNs in BAL")
ggsave(levinrobustBAL, filename = "IFNs_inBAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)
ifnBAL

ifnPBMC <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(ifngenes,rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:(12 + length(intersect(ifngenes,rownames(analysis$ddsPBMCp12))))), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap( ~ gene, scales = "free_y") + scale_y_log10() + theme_bw() + labs(title = "IFNs in PBMC")
ggsave(levinrobustPBMC, filename = "IFNs_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)
ifnPBMC

grep("IFITM", rownames(analysis$ddsBALp12), value = TRUE)
linePlotsGKT(analysis$ddsBALp12, "IFITM5", "SubjectID", "Timepoint", "Treatment", main = "IFITM5 in BAL") + scale_y_log10()
ifitm5PBMC <- linePlotsGKT(analysis$ddsPBMCp12, "IFITM5", "SubjectID", "Timepoint", "Treatment", main = "IFITM5 in PBMC") + scale_y_log10()

ggsave(ifitm5PBMC, filename = "IFITM5_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)
ifitm5PBMC

plotgene = "OAS1"
as.data.frame(analysis$TimepointTreatmentMeansPBMC) %>% rownames_to_column("gene") %>% pivot_longer(-gene,names_to = "TimepointTreatment",values_to = "mean_normalized_count") %>% mutate(Timepoint = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",1)), Treatment = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",2))) %>% filter(gene == plotgene) %>% ggplot(aes(x = Timepoint, y = mean_normalized_count)) + geom_line(aes(color = Treatment, group = Treatment)) + scale_y_log10() + theme_bw() + labs(title = paste0(plotgene, " in PBMC")) + scale_color_manual(values = c("blue","red")) + facet_wrap(~ gene, scales = "free_y")

as.data.frame(analysis$TimepointTreatmentMeansPBMC) %>% rownames_to_column("gene") %>% pivot_longer(-gene,names_to = "TimepointTreatment",values_to = "mean_normalized_count") %>% mutate(Timepoint = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",1)), Treatment = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",2))) %>% filter(gene %in% LevinList$Robust) %>% ggplot(aes(x = Timepoint, y = mean_normalized_count)) + geom_line(aes(color = Treatment, group = Treatment)) + scale_y_log10() + theme_bw() + scale_color_manual(values = c("blue","red")) + facet_wrap(~ gene, scales = "free_y") + labs(title = "Robust ISGs in PBMC") 


as.data.frame(analysis$TimepointTreatmentMeansPBMC) %>% rownames_to_column("gene") %>% pivot_longer(-gene,names_to = "TimepointTreatment",values_to = "mean_normalized_count") %>% mutate(Timepoint = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",1)), Treatment = unlist(lapply(str_split(TimepointTreatment,'\\.'),"[",2))) %>% filter(gene %in% LevinList$Robust) %>% ggplot(aes(x = Treatment, y = mean_normalized_count)) + geom_line(aes(group = gene)) + scale_y_log10() + theme_bw() + scale_color_manual(values = c("blue","red")) + facet_wrap(~ Timepoint, scales = "free_y") + labs(title = "Robust ISGs in PBMC")

```


## GSEA results
### Day2 vs BL in INFmod Infected
#### HALLMARK_IL6_JAK_STAT3_SIGNALING

```{r}

resGSEA_BAL_D2vsBL_IFNmod <- resultsGSEArun(gsearundir = "GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/","Day2 vs BL in INFmod Infected BAL")

resGSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT <- resultsGSEAset(
  "GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",
  "HALLMARK_IL6_JAK_STAT3_SIGNALING")

leGenes_GSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT <- filter(resGSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT,`CORE ENRICHMENT (IFNant_Day2vsIFNant_BL)` == "Yes") %>% pull(SYMBOL)


resGSEA_BAL_D2vsBL_Ctrl <- resultsGSEArun(gsearundir = "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/","Day2 vs BL in Control Infected BAL")

resGSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT <- resultsGSEAset(
  "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",
  "HALLMARK_IL6_JAK_STAT3_SIGNALING")

leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT <- filter(resGSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,`CORE ENRICHMENT (Untreated_Day2vsUntreated_BL)` == "Yes") %>% pull(SYMBOL)


```

```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_leGenes_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 6))
draw(ht_BAL_Ctrl_leGenes_D2vsBL_IL6JAKSTAT)



```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 6))
draw(ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT, column_title = "leGenes_D2vsBL_IL6JAKSTAT in BAL")



```

```{r fig.height=5, fig.width=11}

ht_BAL_UT_leGenes_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALUT_FromMean[rownames(analysis$rlogBALUT_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_IL6JAKSTAT,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 6))
draw(ht_BAL_UT_leGenes_D2vsBL_IL6JAKSTAT)

#ht_BAL_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs, column_title = "ISGs in UT BAL")), ncol = 1, nrow = 1)
#ggsave(ht_BAL_ISGs_UT_layout, filename = "ISG_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)



```




```{r}
ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT, column_title = "leGenes_D2vsBL_IL6JAKSTAT in BAL"))
ht_BAL_Ctrl_leGenes_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenes_D2vsBL_IL6JAKSTAT))
ht_BAL_UT_leGenes_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_UT_leGenes_D2vsBL_IL6JAKSTAT))
ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT_layout <- ggarrange(ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT_plot, ht_BAL_Ctrl_leGenes_D2vsBL_IL6JAKSTAT_plot, ncol = 1, nrow = 3,
                                       ggarrange(ht_BAL_UT_leGenes_D2vsBL_IL6JAKSTAT_plot, ncol = 2, nrow = 1, widths = c(3,1)))
ggsave(ht_BAL_IFNant_leGenes_D2vsBL_IL6JAKSTAT_layout, filename = "IL6JAKSTAT_BAL_Heatmaps.pdf", units = "in", height = 11, width = 8.5, scale = 1)


```



```{r}

## for manuscript

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,20),analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 24, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(5,"mm"), column_title_gp = gpar(fontsize = 18), col = heatscale4, row_names_gp = gpar(fontsize = 16), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,20),analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 24, col = "#FA8072"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(5,"mm"), column_title_gp = gpar(fontsize = 12), col = heatscale4, row_names_gp = gpar(fontsize = 16), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT, column_title = "BAL - IL6 JAK/STAT"))

ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT))

ht_BAL_leGenesCtrl_D2vsBL_IL6JAKSTAT_layout <- ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot, ncol = 1, heights = c(1.14,1))

ggsave(ht_BAL_leGenesCtrl_D2vsBL_IL6JAKSTAT_layout, filename = "IL6JAKSTAT_BAL_Fig_Heatmaps.pdf", units = "in", height = 11, width = 8.5, scale = 1)

```

```{r}

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,20),analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)


ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,20),analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT, column_title = "BAL - IL6 JAK/STAT"))

ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT))

ht_BAL_leGenesCtrl_D2vsBL_IL6JAKSTAT_layout <- ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot, ncol = 1, heights = c(1.36,1))

ggsave(ht_BAL_leGenesCtrl_D2vsBL_IL6JAKSTAT_layout, filename = "IL6JAKSTAT_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 4, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,10),analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "IL6JAKSTAT_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_IL6JAKSTAT,10),analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "IL6JAKSTAT_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)


```

### ISGs

```{r}
## manuscript
## fig 5A ISGs

resGSEA_BAL_D2vsBL_Ctrl_ISG <- resultsGSEAset(
  "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",
  "ISG")

leGenes_GSEA_BAL_D2vsBL_Ctrl_ISG <- filter(resGSEA_BAL_D2vsBL_Ctrl_ISG,`CORE ENRICHMENT (Untreated_Day2vsUntreated_BL)` == "Yes") %>% pull(SYMBOL)

## Don't use leading edge just adopt the order in the current control heatmap
ISG_genes_BAL_order <- c("CXCL10","ISG15","IRF7","MX2","IFI27","RSAD2","IFIT5","IFIT2","IFIT3","OAS1","ISG20","OAS3","OAS2","MX1","IFI44L","IFI6","STAT1")

# ht_BAL_Ctrl_IGSs <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% ISGList,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
# draw(ht_BAL_Ctrl_IGSs)

# ht_BAL_IFNant_ISGs <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% ISGList,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4)
# draw(ht_BAL_IFNant_ISGs, column_title = "ISGs in BAL")

# ht_BAL_IFNant_ISGs_plot <- grid.grabExpr(draw(ht_BAL_IFNant_ISGs, column_title = "ISGs in BAL"))
# ht_BAL_Ctrl_IGSs_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_IGSs))
# ht_BAL_ISGs_layout <- ggarrange(ht_BAL_IFNant_ISGs_plot, ht_BAL_Ctrl_IGSs_plot, ncol = 1, nrow = 2)
# ggsave(ht_BAL_ISGs_layout, filename = "ISG_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)

ht_BAL_IFNmod_ISG_Ctrl_order <- Heatmap(analysis$rlogBALp12_FromMean[ISG_genes_BAL_order,analysis$rldBALp12$Treatment == "IFNant"],
        name = "log2FoldDiff\nfrom Mean",
        row_title = "IFNmod",
        row_title_gp = gpar(fontsize = 12, col = "#003C5F"),
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"),
#        column_title = NULL,
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        column_title_gp = gpar(fontsize = 6),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE
        )

ht_BAL_IFNmod_ISG_Ctrl_order_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_ISG_Ctrl_order,column_title = "BAL-ISGs"))

ggsave(ht_BAL_IFNmod_ISG_Ctrl_order_plot, filename = "ISG_BAL_IFNmod_Fig_Heatmap_small.pdf", units = "in", height = 2, width = 3, scale = 1)
ht_BAL_IFNmod_ISG_Ctrl_order


# analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% ISGList,analysis$rldBALp12$Treatment == "Untreated"]

as.data.frame(analysis$rlogBALp12_FromMean[ISG_genes_BAL_order[-1],analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "ISG_BAL_Untreated_Fig_Heatmap_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[ISG_genes_BAL_order[-1],analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "ISG_BAL_IFNmod_Fig_Heatmap_table.xlsx", colNames = TRUE)


```



#### HALLMARK_TNFA_SIGNALING_VIA_NFKB

```{r}

resGSEA_BAL_D2vsBL_IFNmod_TNFA <- resultsGSEAset(
  "GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",
  "HALLMARK_TNFA_SIGNALING_VIA_NFKB")

leGenes_GSEA_BAL_D2vsBL_IFNmod_TNFA <- filter(resGSEA_BAL_D2vsBL_IFNmod_TNFA,`CORE ENRICHMENT (IFNant_Day2vsIFNant_BL)` == "Yes") %>% pull(SYMBOL)


# resGSEA_BAL_D2vsBL_Ctrl <- resultsGSEArun(gsearundir = "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/","Day2 vs BL in Control Infected BAL")

resGSEA_BAL_D2vsBL_Ctrl_TNFA <- resultsGSEAset(
  "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",
  "HALLMARK_TNFA_SIGNALING_VIA_NFKB")

leGenes_GSEA_BAL_D2vsBL_Ctrl_TNFA <- filter(resGSEA_BAL_D2vsBL_Ctrl_TNFA,`CORE ENRICHMENT (Untreated_Day2vsUntreated_BL)` == "Yes") %>% pull(SYMBOL)

```

```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_leGenes_D2vsBL_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_TNFA,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 3))
draw(ht_BAL_Ctrl_leGenes_D2vsBL_TNFA)



```


```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_leGenes_D2vsBL_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_TNFA,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 3))
draw(ht_BAL_IFNant_leGenes_D2vsBL_TNFA, column_title = "leGenes_D2vsBL_TNFA in BAL")



```


```{r fig.height=5, fig.width=11}

ht_BAL_UT_leGenes_D2vsBL_TNFA <- Heatmap(analysis$rlogBALUT_FromMean[rownames(analysis$rlogBALUT_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_TNFA,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 3))
draw(ht_BAL_UT_leGenes_D2vsBL_TNFA)

#ht_BAL_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs, column_title = "ISGs in UT BAL")), ncol = 1, nrow = 1)
#ggsave(ht_BAL_ISGs_UT_layout, filename = "ISG_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)



```


```{r}
ht_BAL_IFNant_leGenes_D2vsBL_TNFA_plot <- grid.grabExpr(draw(ht_BAL_IFNant_leGenes_D2vsBL_TNFA, column_title = "leGenes_D2vsBL_TNFA in BAL"))
ht_BAL_Ctrl_leGenes_D2vsBL_TNFA_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenes_D2vsBL_TNFA))
ht_BAL_UT_leGenes_D2vsBL_TNFA_plot <- grid.grabExpr(draw(ht_BAL_UT_leGenes_D2vsBL_TNFA))
ht_BAL_IFNant_leGenes_D2vsBL_TNFA_layout <- ggarrange(ht_BAL_IFNant_leGenes_D2vsBL_TNFA_plot, ht_BAL_Ctrl_leGenes_D2vsBL_TNFA_plot, ncol = 1, nrow = 3,
                                       ggarrange(ht_BAL_UT_leGenes_D2vsBL_TNFA_plot, ncol = 2, nrow = 1, widths = c(3,1)))
ggsave(ht_BAL_IFNant_leGenes_D2vsBL_TNFA_layout, filename = "TNFA_BAL_Heatmaps.pdf", units = "in", height = 11, width = 8.5, scale = 1)


```

```{r}

# for manuscript small

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_TNFA,20),analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)


ht_BAL_Ctrl_leGenesCtrl_D2vsBL_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_TNFA,20),analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_TNFA_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_TNFA, column_title = "BAL - TNF"))

ht_BAL_Ctrl_leGenesCtrl_D2vsBL_TNFA_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenesCtrl_D2vsBL_TNFA))

ht_BAL_leGenesCtrl_D2vsBL_TNFA_layout <- ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_TNFA_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_TNFA_plot, ncol = 1, heights = c(1.36,1))

ggsave(ht_BAL_leGenesCtrl_D2vsBL_TNFA_layout, filename = "TNFA_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 4, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_TNFA,10),analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "TNFA_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_TNFA,10),analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "TNFA_BAL_Fig_Heatmaps_INFmod_table.xlsx", colNames = TRUE)


```



#### REACTOME_NEUTROPHIL_DEGRANULATION

```{r}

resGSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN <- resultsGSEAset(
  "GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",
  "REACTOME_NEUTROPHIL_DEGRANULATION")

leGenes_GSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN <- filter(resGSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN,`CORE ENRICHMENT (IFNant_Day2vsIFNant_BL)` == "Yes") %>% pull(SYMBOL)


resGSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN <- resultsGSEAset(
  "GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",
  "REACTOME_NEUTROPHIL_DEGRANULATION")

leGenes_GSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN <- filter(resGSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN,`CORE ENRICHMENT (Untreated_Day2vsUntreated_BL)` == "Yes") %>% pull(SYMBOL)

```

```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_leGenes_D2vsBL_NEUTDEGRAN <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 4))
draw(ht_BAL_Ctrl_leGenes_D2vsBL_NEUTDEGRAN)



```


```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 4))
draw(ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN, column_title = "leGenes_D2vsBL_NEUTDEGRAN in BAL")



```


```{r fig.height=5, fig.width=11}

ht_BAL_UT_leGenes_D2vsBL_NEUTDEGRAN <- Heatmap(analysis$rlogBALUT_FromMean[rownames(analysis$rlogBALUT_FromMean) %in% leGenes_GSEA_BAL_D2vsBL_IFNmod_NEUTDEGRAN,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 4))
draw(ht_BAL_UT_leGenes_D2vsBL_NEUTDEGRAN)

#ht_BAL_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs, column_title = "ISGs in UT BAL")), ncol = 1, nrow = 1)
#ggsave(ht_BAL_ISGs_UT_layout, filename = "ISG_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)



```


```{r}
ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN_plot <- grid.grabExpr(draw(ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN, column_title = "leGenes_D2vsBL_NEUTDEGRAN in BAL"))
ht_BAL_Ctrl_leGenes_D2vsBL_NEUTDEGRAN_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenes_D2vsBL_NEUTDEGRAN))
ht_BAL_UT_leGenes_D2vsBL_NEUTDEGRAN_plot <- grid.grabExpr(draw(ht_BAL_UT_leGenes_D2vsBL_NEUTDEGRAN))
ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN_layout <- ggarrange(ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN_plot, ht_BAL_Ctrl_leGenes_D2vsBL_NEUTDEGRAN_plot, ncol = 1, nrow = 3,
                                       ggarrange(ht_BAL_UT_leGenes_D2vsBL_NEUTDEGRAN_plot, ncol = 2, nrow = 1, widths = c(3,1)))
ggsave(ht_BAL_IFNant_leGenes_D2vsBL_NEUTDEGRAN_layout, filename = "NEUTDEGRAN_BAL_Heatmaps.pdf", units = "in", height = 11, width = 8.5, scale = 1)


```



```{r}

# for manuscript small

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_NEUTDEGRAN <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN,20),analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)


ht_BAL_Ctrl_leGenesCtrl_D2vsBL_NEUTDEGRAN <- Heatmap(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN,20),analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_NEUTDEGRAN, column_title = "BAL - Neut Degran"))

ht_BAL_Ctrl_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_leGenesCtrl_D2vsBL_NEUTDEGRAN))

ht_BAL_leGenesCtrl_D2vsBL_NEUTDEGRAN_layout <- ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot, ncol = 1, heights = c(1.36,1))

ggsave(ht_BAL_leGenesCtrl_D2vsBL_NEUTDEGRAN_layout, filename = "NEUTDEGRAN_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 4, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN,10),analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "NEUTDEGRAN_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[ head(leGenes_GSEA_BAL_D2vsBL_Ctrl_NEUTDEGRAN,10),analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "NEUTDEGRAN_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)

```

```{r}

ht_BAL_leGenesCtrl_D2vsBL_SetsOfInterest_layout <- ggarrange(ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_IL6JAKSTAT_plot, ncol = 1, heights = c(1.36,1)),ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_TNFA_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_TNFA_plot, ncol = 1, heights = c(1.36,1)),ggarrange(ht_BAL_IFNmod_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot,ht_BAL_Ctrl_leGenesCtrl_D2vsBL_NEUTDEGRAN_plot, ncol = 1, heights = c(1.36,1)),nrow = 1, ncol = 3)

ggsave(ht_BAL_leGenesCtrl_D2vsBL_SetsOfInterest_layout, filename = "BAL_Fig_Heatmaps_small.pdf", units = "in", height = 4, width = 9.5, scale = 1)

```

## Steve's table of genes

```{r}

manuscript_genes_of_interest_table <- read.xlsx("Genes_of_interest.xlsx")
inflam_List <- read.xlsx("Genes_of_interest.xlsx", cols = c(1), startRow = 1, rows = c(1,3:10))
other_inflam_List <- read.xlsx("Genes_of_interest.xlsx", cols = c(1), startRow = 12, rows = c(12:17))
chemokine_List <- read.xlsx("Genes_of_interest.xlsx", cols = c(1), startRow = 21, rows = c(21:33))
t_activation_List <- read.xlsx("Genes_of_interest.xlsx", cols = c(1), startRow = 35, rows = c(35:41))

```

```{r}

ht_BAL_Ctrl_inflam <- Heatmap(analysis$rlogBALp12_FromMean[inflam_List$Inflammasome.Cluster,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), show_row_dend = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

inflam_IFNmod_order = c("IL27","IL1RN","IL1B","IL18","NLRP3","GSDMD","CASP1","AIM2")

ht_BAL_IFNmod_inflam <- Heatmap(analysis$rlogBALp12_FromMean[inflam_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_inflam_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_inflam, column_title = "BAL - Inflammasome"))

ht_BAL_Ctrl_inflam_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_inflam))

ht_BAL_inflam_layout <- ggarrange(ht_BAL_IFNmod_inflam_plot,ht_BAL_Ctrl_inflam_plot, ncol = 1, heights = c(1.75,1))

ggsave(ht_BAL_inflam_layout, filename = "Inflammasome_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 2.25, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[inflam_List$Inflammasome.Cluster,analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Inflammasome_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[inflam_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Inflammasome_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)

```



```{r}

ht_BAL_Ctrl_other_inflam <- Heatmap(analysis$rlogBALp12_FromMean[other_inflam_List$Other.inflammatory,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), show_row_dend = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

other_inflam_IFNmod_order = c("IL10","TNF","IFNG","IL7","IL15")

ht_BAL_IFNmod_other_inflam <- Heatmap(analysis$rlogBALp12_FromMean[other_inflam_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_other_inflam_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_other_inflam, column_title = "BAL - Other_inflammatory"))

ht_BAL_Ctrl_other_inflam_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_other_inflam))

ht_BAL_other_inflam_layout <- ggarrange(ht_BAL_IFNmod_other_inflam_plot,ht_BAL_Ctrl_other_inflam_plot, ncol = 1, heights = c(2.1,1))

ggsave(ht_BAL_other_inflam_layout, filename = "Other_inflammatory_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 1.75, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[other_inflam_List$Other.inflammatory,analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Other_inflammatory_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[other_inflam_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Other_inflammatory_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)


```



```{r}

chemokine_order = c("CCL2","CCL3","CCL4L1","CCL5","CCL7","CCL8","CCL20","CCL22","CXCL3","CXCL9","CXCL10","CXCL11")

ht_BAL_Ctrl_chemokine <- Heatmap(analysis$rlogBALp12_FromMean[chemokine_order,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

chemokine_IFNmod_order = c("IL10","TNF","IFNG","IL7","IL15")

ht_BAL_IFNmod_chemokine <- Heatmap(analysis$rlogBALp12_FromMean[chemokine_order,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_chemokine_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_chemokine, column_title = "BAL - Chemokines"))

ht_BAL_Ctrl_chemokine_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_chemokine))

ht_BAL_chemokine_layout <- ggarrange(ht_BAL_IFNmod_chemokine_plot,ht_BAL_Ctrl_chemokine_plot, ncol = 1, heights = c(1.5,1))

ggsave(ht_BAL_chemokine_layout, filename = "Chemokines_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 2.75, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[chemokine_order,analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Chemokines_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[chemokine_order,analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "Chemokines_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)

```



```{r}

ht_BAL_Ctrl_t_activation <- Heatmap(analysis$rlogBALp12_FromMean[t_activation_List$T.cell.activation,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", row_title_gp = gpar(fontsize = 12, col = "#FA8072"), show_row_dend = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE, column_title = NULL,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

t_activation_IFNmod_order = c("CD38","CD274","CD80","CD86","CD69","CD40")

ht_BAL_IFNmod_t_activation <- Heatmap(analysis$rlogBALp12_FromMean[t_activation_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
#                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
                                                     clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)

ht_BAL_IFNmod_t_activation_plot <- grid.grabExpr(draw(ht_BAL_IFNmod_t_activation, column_title = "BAL - T cell activation"))

ht_BAL_Ctrl_t_activation_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_t_activation))

ht_BAL_t_activation_layout <- ggarrange(ht_BAL_IFNmod_t_activation_plot,ht_BAL_Ctrl_t_activation_plot, ncol = 1, heights = c(1.9,1))

ggsave(ht_BAL_t_activation_layout, filename = "T_Activation_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 1.9, width = 3, scale = 1)


as.data.frame(analysis$rlogBALp12_FromMean[t_activation_List$T.cell.activation,analysis$rldBALp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "T_Activation_BAL_Fig_Heatmaps_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogBALp12_FromMean[t_activation_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "T_Activation_BAL_Fig_Heatmaps_IFNmod_table.xlsx", colNames = TRUE)

```

```{r}

ht_BAL_GOI_layout <- ggarrange(
  ggarrange(ggplot(),ht_BAL_IFNmod_inflam_plot,ht_BAL_Ctrl_inflam_plot, ncol = 1, heights = c(0.6,1.75,1)),
  ggarrange(ggplot(),ht_BAL_IFNmod_other_inflam_plot,ht_BAL_Ctrl_other_inflam_plot, ncol = 1, heights = c(1.75,2.1,1)),
  ggarrange(ht_BAL_IFNmod_chemokine_plot,ht_BAL_Ctrl_chemokine_plot, ncol = 1, heights = c(1.5,1)),
  ggarrange(ggplot(),ht_BAL_IFNmod_t_activation_plot,ht_BAL_Ctrl_t_activation_plot, ncol = 1, heights = c(1.25,1.9,1)),nrow = 1, ncol = 4)

ggsave(ht_BAL_GOI_layout, filename = "GOI_BAL_Fig_Heatmaps_small.pdf", units = "in", height = 2.75, width = 12, scale = 1)

```




## Tim-20220218 List
```{r fig.height=8.5, fig.width=11}

Tim20220218List <- read_lines("Tim-20220218_List.txt",skip_empty_rows = TRUE)



```

```{r fig.height=5, fig.width=11}

ht_Ctrl_Tim20220218 <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Tim20220218List,analysis$rldPBMCp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Control", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_Ctrl_Tim20220218)

```


```{r fig.height=5, fig.width=11}

ht_IFNant_Tim20220218 <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Tim20220218List,analysis$rldPBMCp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "IFNant", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4)
draw(ht_IFNant_Tim20220218, column_title = "Tim-20220218 in PBMC")

```


```{r}

ht_IFNant_Tim20220218_plot <- grid.grabExpr(draw(ht_IFNant_Tim20220218, column_title = "Tim-20220218 in PBMC"))
ht_Ctrl_Tim20220218_plot <- grid.grabExpr(draw(ht_Ctrl_Tim20220218))
ht_Tim20220218_layout <- ggarrange(ht_IFNant_Tim20220218_plot, ht_Ctrl_Tim20220218_plot, ncol = 1, nrow = 2)
ggsave(ht_Tim20220218_layout, filename = "Tim-20220218_PBMC_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```



```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_Tim20220218 <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Tim20220218List,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_BAL_Ctrl_Tim20220218)



```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_Tim20220218 <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Tim20220218List,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_IFNant_Tim20220218, column_title = "Tim-20220218 in BAL")



```


```{r}

ht_BAL_IFNant_Tim20220218_plot <- grid.grabExpr(draw(ht_BAL_IFNant_Tim20220218, column_title = "Tim-20220218 in BAL"))
ht_BAL_Ctrl_Tim20220218_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_Tim20220218))
ht_BAL_Tim20220218_layout <- ggarrange(ht_BAL_IFNant_Tim20220218_plot, ht_BAL_Ctrl_Tim20220218_plot, ncol = 1, nrow = 2)
ggsave(ht_BAL_Tim20220218_layout, filename = "Tim-20220218_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```




```{r fig.height=8.5, fig.width=11}

## Phase I and II Fold Change from Baseline plots of genes of interest

#cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggplot(aes(x = Timepoint, y=lFCfromBL)) + geom_boxplot(aes(group = interaction(Timepoint,Treatment), color = Treatment), outlier.shape = NA) + geom_sina( aes(group = interaction(Timepoint,Treatment), color = Treatment)) + 
#  geom_text_repel(aes(group = interaction(Timepoint,Treatment),label = SubjectID), size = 3) +
#  theme_bw() +
#  facet_wrap(~ gene, scales = "free_y")

robustPBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Robust ISG Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(robustPBMClFC, filename = "Robust_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

#robustPBMClFCMSD <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Robust ISG Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted")
#ggsave(robustPBMClFCMSE, filename = "Robust_PBMC_log2_FoldChange_MSD.pdf", units = "in", width = 11, height = 8.5, scale = 1)

robustBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Robust ISG Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(robustBALlFC, filename = "Robust_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

tunablePBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% tunable) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Tunable ISG Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(tunablePBMClFC, filename = "Tunable_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

tunableBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% tunable) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Tunable ISG Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(tunableBALlFC, filename = "Tunable_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

enhancingPBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% enhancing) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Enhancing Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(enhancingPBMClFC, filename = "Enhancing_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

enhancingBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% enhancing) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter",add.params = list(size = 1, alpha = 0.5, width = 0.6),outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Enhancing Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(enhancingBALlFC, filename = "Enhancing_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

inhibitoryPBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% inhibitory) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Inhibitory Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(inhibitoryPBMClFC, filename = "Inhibitory_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

inhibitoryBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% inhibitory) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter",add.params = list(size = 1, alpha = 0.5, width = 0.6),outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Inhibitory Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(inhibitoryBALlFC, filename = "Inhibitory_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

isglistPBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% ISGList) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "ISGlist Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(isglistPBMClFC, filename = "ISGList_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

isglistBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% ISGList) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "ISGlist Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFant"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(isglistBALlFC, filename = "ISGList_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

# for manuscript
# IRF7

IRF7BALlFC <- cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF7")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA) + theme_bw() + theme(axis.text = element_text(color = "black")) + labs(title = "IRF7 Fold Changes in BAL") + ylab(label = expression(Log[2]~Fold-change~from~baseline)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFmod"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "2 dpi","Day4" = "4 dpi", "Day5" = "D5", "Day7" = "7 dpi"))

ggsave(IRF7BALlFC, filename = "IRF7_BAL_log2_FoldChange.pdf", units = "in", width = 4, height = 3, scale = 1)
ggsave(IRF7BALlFC, filename = "IRF7_BAL_log2_FoldChange.svg", units = "in", width = 4, height = 3, scale = 1)

# ISG15

ISG15BALlFC <- cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("ISG15")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA) + theme_bw() + theme(axis.text = element_text(color = "black")) +labs(title = "ISG15 Fold Changes in BAL") + ylab(label = expression(Log[2]~Fold-change~from~baseline)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFmod"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "2 dpi","Day4" = "4 dpi", "Day5" = "D5", "Day7" = "7 dpi"))

ggsave(ISG15BALlFC, filename = "ISG15_BAL_log2_FoldChange.pdf", units = "in", width = 4, height = 3, scale = 1)
ggsave(ISG15BALlFC, filename = "ISG15_BAL_log2_FoldChange.svg", units = "in", width = 4, height = 3, scale = 1)

# ISG20

ISG20BALlFC <- cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("ISG20")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA) + theme_bw() + theme(axis.text = element_text(color = "black")) + labs(title = "ISG20 Fold Changes in BAL") + ylab(label = expression(Log[2]~Fold-change~from~baseline)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFmod"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "2 dpi","Day4" = "4 dpi", "Day5" = "D5", "Day7" = "7 dpi"))

ggsave(ISG20BALlFC, filename = "ISG20_BAL_log2_FoldChange.pdf", units = "in", width = 4, height = 3, scale = 1)
ggsave(ISG20BALlFC, filename = "ISG20_BAL_log2_FoldChange.svg", units = "in", width = 4, height = 3, scale = 1)

ggsave(ggarrange(IRF7BALlFC,ISG15BALlFC,ISG20BALlFC,ncol = 1), filename = "BAL_3_log2_FoldChange.pdf", units = "in", width = 4, height = 9, scale = 1)

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF7","ISG15","ISG20")) %>%
  filter(!is.na(lFCfromBL)) %>%
  write.xlsx(file = "BAL_3_log2_FoldChange_table.xlsx", colNames = TRUE)


ISGBALsuplFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% setdiff(ISGList,c("IRF7","ISG15","ISG20"))) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + theme(axis.text = element_text(color = "black")) + labs(title = "ISG Fold Changes in BAL") + ylab(label = expression(Log[2]~Fold-change~from~baseline)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFmod"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "2 dpi","Day4" = "4 dpi", "Day5" = "D5", "Day7" = "7 dpi"))
ggsave(ISGBALsuplFC, filename = "ISG_sup_BAL_log2_FoldChange.pdf", units = "in", width = 8.5, height = 11, scale = 1)

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IFI27","IFI44L","MX2","OAS1","OAS2","IFI6","MX1","OAS3","RSAD2","STAT1")) %>%
  filter(!is.na(lFCfromBL)) %>%
  write.xlsx("ISG_sup_BAL_log2_FoldChange_finalSet.xlsx", colNames = TRUE)

## Save BAL fold change data for Elise to re-plot

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF7","ISG15","ISG20")) %>% write_tsv(path = "Fig_FoldChange_data.tsv")

## Try putting in pvalue indication

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("ISG20")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA) + theme_bw() + theme(axis.text = element_text(color = "black")) + labs(title = "ISG20 Fold Changes in BAL") + ylab(label = expression(Log[2]~Fold-change~from~baseline)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Control","INFmod"), values = c("#FA8072","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "2 dpi","Day4" = "4 dpi", "Day5" = "D5", "Day7" = "7 dpi")) + stat_compare_means(comparisons = list(c(1,5)), aes(label = "0.0001"))


rebuttalBALlFC <- facet(cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter",add.params = list(size = 1, alpha = 0.5, width = 0.6),outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Rebuttal Set Fold Changes in BAL") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated","INFmod"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(rebuttalBALlFC, filename = "Rebuttal_BAL_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

cbind(colData(analysis$ddsBALp12),t(assay(analysis$rldBALp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay2 = Day2-BL, lFCDay4 = Day4-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11")) %>%
  filter(!is.na(lFCfromBL)) %>%
  write.xlsx(file = "Rebuttal_BAL_log2_FoldChange_table.xlsx", colNames = TRUE)


rebuttalPBMClFC <- facet(cbind(colData(analysis$ddsPBMCp12),t(assay(analysis$rldPBMCp12))) %>% as.data.frame() %>% pivot_longer(-c(1:12), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, lFCDay4 = Day4-BL,  lFCDay5 = Day5-BL, lFCDay7 = Day7-BL, .keep = "unused") %>% pivot_longer(c(4:9), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Rebuttal Set Fold Changes in PBMC") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated","INFmod"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(rebuttalPBMClFC, filename = "Rebuttal_PBMC_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

rebuttalPBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Rebuttal Set Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Uninfected-INFmod"), values = c("#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(rebuttalPBMCUTlFC, filename = "Rebuttal_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1")) %>%
  write.xlsx("Rebuttal_PBMC_UT_log2_FoldChange_IRF1_table.xlsx", colNames = TRUE)

rebuttalBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11")) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Rebuttal Set Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Uninfected-INFmod"), values = c("#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(rebuttalBALUTlFC, filename = "Rebuttal_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% c("IRF1")) %>%
  write.xlsx("Rebuttal_BAL_UT_log2_FoldChange_IRF1_table.xlsx", colNames = TRUE)

rebuttalBALline <- cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11"),rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:18), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene, scales = "free_y") + scale_y_log10() + scale_color_manual(labels = c("Untreated","INFmod"), values = c("#F73019","#003C5F")) + theme_bw()
ggsave(rebuttalBALline, filename = "Rebuttal_LinePlots_inBAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)

rebuttalPBMCline <- cbind(colData(analysis$ddsPBMCp12),t(counts(analysis$ddsPBMCp12, normalized = TRUE)[intersect(c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11"),rownames(analysis$ddsPBMCp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:18), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene, scales = "free_y") + scale_y_log10() + scale_color_manual(labels = c("Untreated","INFmod"), values = c("#F73019","#003C5F")) + theme_bw()
ggsave(rebuttalPBMCline, filename = "Rebuttal_LinePlots_inPBMC.pdf", units = "in", width = 11, height = 8.5, scale = 1)

rebuttalBALUTline <- cbind(colData(analysis$ddsBALUT),t(counts(analysis$ddsBALUT, normalized = TRUE)[intersect(c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11"),rownames(analysis$ddsBALUT)),])) %>% as.data.frame() %>% pivot_longer( c(11:16), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene, scales = "free_y") + scale_y_log10() + scale_color_manual(labels = c("Uninfected-INFmod"), values = c("#003C5F")) + theme_bw()
ggsave(rebuttalBALUTline, filename = "Rebuttal_LinePlots_inBAL_UT.pdf", units = "in", width = 11, height = 8.5, scale = 1)

rebuttalPBMCUTline <- cbind(colData(analysis$ddsPBMCUT),t(counts(analysis$ddsPBMCUT, normalized = TRUE)[intersect(c("IRF1","IRF7","IRF9","CXCL9","CXCL10","CXCL11"),rownames(analysis$ddsPBMCUT)),])) %>% as.data.frame() %>% pivot_longer( c(11:16), names_to = "gene", values_to = "count") %>% ggplot(aes(x = Timepoint, y=count)) + geom_line(aes(color=Treatment, group = SubjectID)) + facet_wrap(~ gene, scales = "free_y") + scale_y_log10() + scale_color_manual(labels = c("Uninfected-INFmod"), values = c("#003C5F")) + theme_bw()
ggsave(rebuttalPBMCUTline, filename = "Rebuttal_LinePlots_inPBMC_UT.pdf", units = "in", width = 11, height = 8.5, scale = 1)




rebuttalBALbox <- facet(cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(c("CLEC4C","IFNA6","IFNA8","IFNA10","IFNA13","IFNA14"),rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:18), names_to = "gene", values_to = "count") %>% filter(Timepoint == "Day2") %>% 
#  ggplot(aes(x = Treatment, y=count)) + geom_boxplot(aes(color=Treatment, group = SubjectID),width = 0.2, outlier.shape = NA) + geom_jitter(aes(color=Treatment), width = 0.1) + facet_wrap(~ gene, scales = "free_y")  + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + theme_bw()
  ggerrorplot(x = "Timepoint", y = "count", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Rebuttal CLEC4C and IFNA in BAL") + ylab(label = "normalized count")  + scale_color_manual(labels = c("Untreated","IFNmod"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day2" = "D2"))
ggsave(rebuttalBALbox, filename = "Rebuttal_boxPlots_Day2_BAL.pdf", units = "in", width = 11, height = 8.5, scale = 1)

cbind(colData(analysis$ddsBALp12),t(counts(analysis$ddsBALp12, normalized = TRUE)[intersect(c("CLEC4C","IFNA6","IFNA8","IFNA10","IFNA13","IFNA14"),rownames(analysis$ddsBALp12)),])) %>% as.data.frame() %>% pivot_longer( c(13:18), names_to = "gene", values_to = "count") %>% filter(Timepoint == "Day2") %>%
  write.xlsx(file = "Rebuttal_boxPlots_Day2_BAL_table.xlsx", colNames = TRUE)

write.xlsx(data.frame(as.data.frame(colData(analysis$ddsBALp12)), CLEC4C = counts(analysis$ddsBALp12, normalized = TRUE)["CLEC4C",]), file = "CLEC4C_BAL_normCounts.xlsx", colNames = TRUE)


```

### Phase III Uninfected Treated

```{r}

# Robust
robustPBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Robust ISG Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(robustPBMCUTlFC, filename = "Robust_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

robustBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% robust) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Robust ISG Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(robustBALUTlFC, filename = "Robust_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

#Tunable
tunablePBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% tunable) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Tunable ISG Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(tunablePBMCUTlFC, filename = "Tunable_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

tunableBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% tunable) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Tunable ISG Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(tunableBALUTlFC, filename = "Tunable_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


#Enhancing
enhancingPBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% enhancing) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Enhancing Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(enhancingPBMCUTlFC, filename = "Enhancing_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


enhancingBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% enhancing) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Enhancing Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(enhancingBALUTlFC, filename = "Enhancing_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


#Inhibitory
inhibitoryPBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% inhibitory) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Inhibitory Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(inhibitoryPBMCUTlFC, filename = "Inhibitory_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


inhibitoryBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% inhibitory) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "Inhibitory Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(inhibitoryBALUTlFC, filename = "Inhibitory_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


#ISGs
isglistPBMCUTlFC <- facet(cbind(colData(analysis$ddsPBMCUT),t(assay(analysis$rldPBMCUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay1 = Day1-BL,  lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:6), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% ISGList) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "ISGList Fold Changes in PBMC UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day1" = "D1","Day2" = "D2"))
ggsave(isglistPBMCUTlFC, filename = "ISGList_PBMC_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)


isglistBALUTlFC <- facet(cbind(colData(analysis$ddsBALUT),t(assay(analysis$rldBALUT))) %>% as.data.frame() %>% pivot_longer(-c(1:10), names_to = "gene", values_to = "rlog") %>% pivot_wider(c("SubjectID","Treatment","gene"), names_from = "Timepoint", values_from = "rlog") %>% mutate(lFCDay0 = Day0-BL, lFCDay2 = Day2-BL, .keep = "unused") %>% pivot_longer(c(4:5), values_to = "lFCfromBL", names_to = "Timepoint", names_prefix = "lFC") %>% filter(gene %in% ISGList) %>% ggerrorplot(x = "Timepoint", y = "lFCfromBL", color = "Treatment", desc_stat = "mean_sd", add = "jitter", add.params = list(size = 1, alpha = 0.5), outlier.shape = NA), facet.by = c("gene"), scales = "free_y") + theme_bw() + labs(title = "ISGList Fold Changes in BAL UT") + ylab(label = expression(log[2]~FoldChange~from~BL)) + geom_hline(yintercept = 0, linetype = "dotted") + scale_color_manual(labels = c("Untreated-INFant"), values = c("#F73019","#003C5F")) + scale_x_discrete(labels = c("Day0" = "D0", "Day2" = "D2","Day4" = "D4", "Day5" = "D5", "Day7" = "D7"))
ggsave(isglistBALUTlFC, filename = "ISGList_BAL_UT_log2_FoldChange.pdf", units = "in", width = 11, height = 8.5, scale = 1)

```

```{r}

merged_BAL_UT_results <- as.annot.data.frame(results(analysis$ddsBALUT,
                                                     contrast = list( c("Timepoint_Day0_vs_BL")),
                                                     alpha = 0.05,
                                                     filter = maxMinFilter(analysis$ddsBALUT,
                                                                           intgroup = "Timepoint",
                                                                           comp = c("Day0","BL"))),
                                             subset = "BAL",
                                             relabel = "Timepoint_Day0_vs_BL effect",
                                             newlabel = "Uninfected 1d tx/-7d tx") %>%
  left_join(as.annot.data.frame(results(analysis$ddsBALUT,
                                        contrast = list( c("Timepoint_Day2_vs_BL")),
                                        alpha = 0.05,
                                        filter = maxMinFilter(analysis$ddsBALUT,
                                                              intgroup = "Timepoint",
                                                              comp = c("Day2","BL"))),
                                subset = "BAL",
                                relabel = "Timepoint_Day2_vs_BL effect",
                                newlabel = "Uninfected 3d tx/-7d tx"))

merged_PBMC_UT_results <- as.annot.data.frame(results(analysis$ddsPBMCUT,
                                                     contrast = list( c("Timepoint_Day0_vs_BL")),
                                                     alpha = 0.05,
                                                     filter = maxMinFilter(analysis$ddsPBMCUT,
                                                                           intgroup = "Timepoint",
                                                                           comp = c("Day0","BL"))),
                                             subset = "PBMC",
                                             relabel = "Timepoint_Day0_vs_BL effect",
                                             newlabel = "Uninfected 1d tx/-7d tx") %>%
  left_join(as.annot.data.frame(results(analysis$ddsPBMCUT,
                                        contrast = list( c("Timepoint_Day1_vs_BL")),
                                        alpha = 0.05,
                                        filter = maxMinFilter(analysis$ddsPBMCUT,
                                                              intgroup = "Timepoint",
                                                              comp = c("Day1","BL"))),
                                subset = "PBMC",
                                relabel = "Timepoint_Day1_vs_BL effect",
                                newlabel = "Uninfected 2d tx/-7d tx")) %>%
  left_join(as.annot.data.frame(results(analysis$ddsPBMCUT,
                                        contrast = list( c("Timepoint_Day2_vs_BL")),
                                        alpha = 0.05,
                                        filter = maxMinFilter(analysis$ddsPBMCUT,
                                                              intgroup = "Timepoint",
                                                              comp = c("Day2","BL"))),
                                subset = "PBMC",
                                relabel = "Timepoint_Day2_vs_BL effect",
                                newlabel = "Uninfected 3d tx/-7d tx"))

merged_UT_results_list <- list("BAL" = arrange(merged_BAL_UT_results,
                                               `pvalue (Wald test p-value: Uninfected 1d tx/-7d tx)`),
                               "PBMC" = arrange(merged_PBMC_UT_results,
                                                `pvalue (Wald test p-value: Uninfected 1d tx/-7d tx)`))

write.xlsx(merged_UT_results_list,
           file = "Phase0_merged_DESeq2_results_COVten.xlsx", colNames = TRUE)


```



## Heatmaps

```{r fig.height=6, fig.width=11}

heatscale2 <- colorRamp2(c(-2,0,2), c("blue","#EEEEEE","red"))
heatscale3 <- colorRamp2(c(-3,0,3), c("blue","#EEEEEE","red"))
heatscale4 <- colorRamp2(c(-4,0,4), c("blue","#EEEEEE","red"))

Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% ISGList,],column_split = interactionPBMCp12_Treatment_Timepoint, cluster_column_slices = FALSE)

```

```{r fig.height=5, fig.width=11}

ht_Ctrl_IGSs <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% ISGList,analysis$rldPBMCp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Control", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_Ctrl_IGSs)

```


```{r fig.height=5, fig.width=11}

ht_IFNant_ISGs <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% ISGList,analysis$rldPBMCp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "IFNant", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4)
draw(ht_IFNant_ISGs, column_title = "ISGs in PBMC")

```


```{r fig.height=5, fig.width=11}

ht_UT_ISGs <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[rownames(analysis$rlogPBMCUT_FromMeanDay0) %in% ISGList,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_UT_ISGs, column_title = "ISGs in PBMC")

ht_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_UT_ISGs, column_title = "ISGs in UT PBMC")), ncol = 1, nrow = 1)
ggsave(ht_ISGs_UT_layout, filename = "ISG_PBMC_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)

```



```{r}

ht_IFNant_ISGs_plot <- grid.grabExpr(draw(ht_IFNant_ISGs, column_title = "ISGs in PBMC"))
ht_Ctrl_IGSs_plot <- grid.grabExpr(draw(ht_Ctrl_IGSs))
ht_ISGs_layout <- ggarrange(ht_IFNant_ISGs_plot, ht_Ctrl_IGSs_plot, ncol = 1, nrow = 2)
ggsave(ht_ISGs_layout, filename = "ISG_PBMC_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```

```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_IGSs <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% ISGList,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_BAL_Ctrl_IGSs)



```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_ISGs <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% ISGList,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4)
draw(ht_BAL_IFNant_ISGs, column_title = "ISGs in BAL")



```

```{r}

ht_BAL_IFNant_ISGs_plot <- grid.grabExpr(draw(ht_BAL_IFNant_ISGs, column_title = "ISGs in BAL"))
ht_BAL_Ctrl_IGSs_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_IGSs))
ht_BAL_ISGs_layout <- ggarrange(ht_BAL_IFNant_ISGs_plot, ht_BAL_Ctrl_IGSs_plot, ncol = 1, nrow = 2)
ggsave(ht_BAL_ISGs_layout, filename = "ISG_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_BAL_UT_ISGs <- Heatmap(analysis$rlogBALUT_FromMean[rownames(analysis$rlogBALUT_FromMean) %in% ISGList,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
draw(ht_BAL_UT_ISGs)

ht_BAL_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs, column_title = "ISGs in UT BAL")), ncol = 1, nrow = 1)
ggsave(ht_BAL_ISGs_UT_layout, filename = "ISG_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)



```



```{r}

Bari_Neut_Degranulation <- c("AZU1","SERPINB10","CLEC4C","CRISPLD2","PRTN3","TNFAIP6","FCN1","EPX","SELL","CXCR1","ADAM8","BPI","QPCT","CXCR2","S100A12","MMP25","MMP9","MGAM","AMPD3","ADGRG3","MPO","SIGLEC5","CFP","C3","LPCAT1","HBB","CD93","LCN2","RHOF","SYNGR1","LRMP","BST1","NFAM1","VNN1","TARM1")
Bari_TNFA <- c("TNFAIP6","GEM","SERPINB2","VEGFA","CXCL6","IER3","MXD1","MARCKS","SPHK1","LAMB3","GPR183","PDE4B","F3","BIRC3","CCL2","IL12B","PTPRE","F2RL1","CXCL3","PTGS2","NFKB2","TNC","FOSL1","CCN1","MSC","SOCS3","SLC2A6","BHLHE40","SLC2A3","RELB","FOSL2","IFNGR2","SOD2","CDKN1A","KLF2")
Bari_IL6_JAK_STAT3 <- c("IL1R2","IL1R1","CXCL13","IL18R1","LTB","CSF3R","OSMR","CXCL3","PIK3R5","SOCS3","CCL7","HMOX1","CD38","IFNGR2","CD14","IFNAR1","IL12RB1","CCR1","TLR2","CD36","IL4R","CSF2","IFNGR1","CXCL9")

```



```{r fig.height=5, fig.width=11}

ht_PBMC_Ctrl_Neut_Degran <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_Neut_Degranulation,analysis$rldPBMCp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Control", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_Ctrl_Neut_Degran)


```

```{r fig.height=5, fig.width=11}

ht_PBMC_IFNant_Neut_Degran <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_Neut_Degranulation,analysis$rldPBMCp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "IFNant", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_IFNant_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in PBMC")


```

```{r}

ht_PBMC_IFNant_Neut_Degran_plot <- grid.grabExpr(draw(ht_PBMC_IFNant_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in PBMC"))
ht_PBMC_Ctrl_Neut_Degran_plot <- grid.grabExpr(draw(ht_PBMC_Ctrl_Neut_Degran))
ht_PBMC_Neut_Degran_layout <- ggarrange(ht_PBMC_IFNant_Neut_Degran_plot, ht_PBMC_Ctrl_Neut_Degran_plot, ncol = 1, nrow = 2)
ggsave(ht_PBMC_Neut_Degran_layout, filename = "Neutrophil_Degranulation_PBMC_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_PBMC_UT_Neut_Degran <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[rownames(analysis$rlogPBMCUT_FromMeanDay0) %in% Bari_Neut_Degranulation,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_UT_Neut_Degran)

ht_PBMC_Neut_Degran_UT_layout <- ggarrange(grid.grabExpr(draw(ht_PBMC_UT_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in UT PBMC")), ncol = 1, nrow = 1)
ggsave(ht_PBMC_Neut_Degran_UT_layout, filename = "Neutrophil_Degranulation_PBMC_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_Neut_Degran <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_Neut_Degranulation,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_Ctrl_Neut_Degran)


```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_Neut_Degran <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_Neut_Degranulation,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_IFNant_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in BAL")


```

```{r}

ht_BAL_IFNant_Neut_Degran_plot <- grid.grabExpr(draw(ht_BAL_IFNant_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in BAL"))
ht_BAL_Ctrl_Neut_Degran_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_Neut_Degran))
ht_BAL_Neut_Degran_layout <- ggarrange(ht_BAL_IFNant_Neut_Degran_plot, ht_BAL_Ctrl_Neut_Degran_plot, ncol = 1, nrow = 2)
ggsave(ht_BAL_Neut_Degran_layout, filename = "Neutrophil_Degranulation_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_BAL_UT_Neut_Degran <- Heatmap(analysis$rlogBALUT_FromMeanDay0[rownames(analysis$rlogBALUT_FromMeanDay0) %in% Bari_Neut_Degranulation,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_UT_Neut_Degran)

ht_BAL_Neut_Degran_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_Neut_Degran, column_title = "NEUTROPHIL DEGRANULATION in UT BAL")), ncol = 1, nrow = 1)
ggsave(ht_BAL_Neut_Degran_UT_layout, filename = "Neutrophil_Degranulation_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```




```{r fig.height=5, fig.width=11}

ht_PBMC_Ctrl_TNFA <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_TNFA,analysis$rldPBMCp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Control", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_Ctrl_TNFA)


```

```{r fig.height=5, fig.width=11}

ht_PBMC_IFNant_TNFA <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_TNFA,analysis$rldPBMCp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "IFNant", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_IFNant_TNFA, column_title = "TNFA SIGNALING in PBMC")


```

```{r}

ht_PBMC_IFNant_TNFA_plot <- grid.grabExpr(draw(ht_PBMC_IFNant_TNFA, column_title = "TNFA SIGNALING in PBMC"))
ht_PBMC_Ctrl_TNFA_plot <- grid.grabExpr(draw(ht_PBMC_Ctrl_TNFA))
ht_PBMC_TNFA_layout <- ggarrange(ht_PBMC_IFNant_TNFA_plot, ht_PBMC_Ctrl_TNFA_plot, ncol = 1, nrow = 2)
ggsave(ht_PBMC_TNFA_layout, filename = "TNFA_SIGNALING_PBMC_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_PBMC_UT_TNFA <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[rownames(analysis$rlogPBMCUT_FromMeanDay0) %in% Bari_TNFA,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_UT_TNFA)

ht_PBMC_TNFA_UT_layout <- ggarrange(grid.grabExpr(draw(ht_PBMC_UT_TNFA, column_title = "TNFA SIGNALING in UT PBMC")), ncol = 1, nrow = 1)
ggsave(ht_PBMC_TNFA_UT_layout, filename = "TNFA_SIGNALING_PBMC_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```



```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_TNFA,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_Ctrl_TNFA)


```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_TNFA <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_TNFA,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_IFNant_TNFA, column_title = "TNFA SIGNALING in BAL")


```

```{r}

ht_BAL_IFNant_TNFA_plot <- grid.grabExpr(draw(ht_BAL_IFNant_TNFA, column_title = "TNFA SIGNALING in BAL"))
ht_BAL_Ctrl_TNFA_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_TNFA))
ht_BAL_TNFA_layout <- ggarrange(ht_BAL_IFNant_TNFA_plot, ht_BAL_Ctrl_TNFA_plot, ncol = 1, nrow = 2)
ggsave(ht_BAL_TNFA_layout, filename = "TNFA_SIGNALING_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_BAL_UT_TNFA <- Heatmap(analysis$rlogBALUT_FromMeanDay0[rownames(analysis$rlogBALUT_FromMeanDay0) %in% Bari_TNFA,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_UT_TNFA)

ht_BAL_TNFA_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_TNFA, column_title = "TNFA SIGNALING in UT BAL")), ncol = 1, nrow = 1)
ggsave(ht_BAL_TNFA_UT_layout, filename = "TNFA_SIGNALING_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_PBMC_Ctrl_IL6_JAK_STAT3 <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Control", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_Ctrl_IL6_JAK_STAT3)


```

```{r fig.height=5, fig.width=11}

ht_PBMC_IFNant_IL6_JAK_STAT3 <- Heatmap(analysis$rlogPBMCp12_FromMeanDay0[rownames(analysis$rlogPBMCp12_FromMeanDay0) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "IFNant", column_split = analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_IFNant_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in PBMC")


```

```{r}

ht_PBMC_IFNant_IL6_JAK_STAT3_plot <- grid.grabExpr(draw(ht_PBMC_IFNant_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in PBMC"))
ht_PBMC_Ctrl_IL6_JAK_STAT3_plot <- grid.grabExpr(draw(ht_PBMC_Ctrl_IL6_JAK_STAT3))
ht_PBMC_IL6_JAK_STAT3_layout <- ggarrange(ht_PBMC_IFNant_IL6_JAK_STAT3_plot, ht_PBMC_Ctrl_IL6_JAK_STAT3_plot, ncol = 1, nrow = 2)
ggsave(ht_PBMC_IL6_JAK_STAT3_layout, filename = "IL6_JAK_STAT3_SIGNALING_PBMC_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```


```{r fig.height=5, fig.width=11}

ht_PBMC_UT_IL6_JAK_STAT3 <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[rownames(analysis$rlogPBMCUT_FromMeanDay0) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_PBMC_UT_IL6_JAK_STAT3)

ht_PBMC_IL6_JAK_STAT3_UT_layout <- ggarrange(grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in UT PBMC")), ncol = 1, nrow = 1)
ggsave(ht_PBMC_IL6_JAK_STAT3_UT_layout, filename = "IL6_JAK_STAT3_SIGNALING_PBMC_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```



```{r fig.height=5, fig.width=11}

ht_BAL_Ctrl_IL6_JAK_STAT3 <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldBALp12$Treatment == "Untreated"],name = "log2FoldDiff\nfrom Mean", row_title = "Control", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_Ctrl_IL6_JAK_STAT3)


```

```{r fig.height=5, fig.width=11}

ht_BAL_IFNant_IL6_JAK_STAT3 <- Heatmap(analysis$rlogBALp12_FromMean[rownames(analysis$rlogBALp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNant", column_split = analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$Timepoint, column_title = NULL, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))), show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_IFNant_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in BAL")


```

```{r}

ht_BAL_IFNant_IL6_JAK_STAT3_plot <- grid.grabExpr(draw(ht_BAL_IFNant_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in BAL"))
ht_BAL_Ctrl_IL6_JAK_STAT3_plot <- grid.grabExpr(draw(ht_BAL_Ctrl_IL6_JAK_STAT3))
ht_BAL_IL6_JAK_STAT3_layout <- ggarrange(ht_BAL_IFNant_IL6_JAK_STAT3_plot, ht_BAL_Ctrl_IL6_JAK_STAT3_plot, ncol = 1, nrow = 2)
ggsave(ht_BAL_IL6_JAK_STAT3_layout, filename = "IL6_JAK_STAT3_SIGNALING_BAL_Heatmaps.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```

```{r fig.height=5, fig.width=11}

ht_BAL_UT_IL6_JAK_STAT3 <- Heatmap(analysis$rlogBALUT_FromMeanDay0[rownames(analysis$rlogBALUT_FromMeanDay0) %in% Bari_IL6_JAK_STAT3,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4, row_names_gp = gpar(fontsize = 7))
draw(ht_BAL_UT_IL6_JAK_STAT3)

ht_BAL_IL6_JAK_STAT3_UT_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in UT BAL")), ncol = 1, nrow = 1)
ggsave(ht_BAL_IL6_JAK_STAT3_UT_layout, filename = "IL6_JAK_STAT3_SIGNALING_BAL_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)


```

## for manuscript

```{r}

# Fig 1C PBMC antiviral ISGs

ISGs_BAL_order <- c("OAS3","OAS1","IFIT5","IRF7","CXCL10","IFIT2","STAT1","RSAD2","ISG20","IFI27","OAS2","MX2","MX1","IFI6","IFIT3","ISG15","IFI44L")

# ht_UT_ISGs <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[rownames(analysis$rlogPBMCUT_FromMeanDay0) %in% ISGList,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],name = "log2FoldDiff\nfrom Mean Day0", row_title = "Uninfected-IFNant", column_split = analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, cluster_column_slices = FALSE, bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))), clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, col = heatscale4)
# draw(ht_UT_ISGs, column_title = "ISGs in PBMC")
# 
# ht_ISGs_UT_layout <- ggarrange(grid.grabExpr(draw(ht_UT_ISGs, column_title = "ISGs in UT PBMC")), ncol = 1, nrow = 1)
# ggsave(ht_ISGs_UT_layout, filename = "ISG_PBMC_UT_Heatmap.pdf", units = "in", height = 8.5, width = 11, scale = 1)

ht_PBMC_UT_ISGs_BAL_order <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[ISGs_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean Day0",
#        row_title = "Uninfected-IFNant",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-7d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_ISGs_BAL_order_plot <- grid.grabExpr(draw(ht_PBMC_UT_ISGs_BAL_order,column_title = "PBMC antiviral ISGs"))

ggsave(ht_PBMC_UT_ISGs_BAL_order_plot, filename = "PBMC_ISGs_Fig_Heatmaps_small.pdf", units = "in", height = 3, width = 3, scale = 1)
ht_PBMC_UT_ISGs_BAL_order


```


Save figure values out in table:
```{r}
analysis$rlogPBMCUT_FromMeanDay0[ISGs_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"][,]
```

### Adjustments to manuscript heatmap baselines

```{r}

## BAL UT meanAll ISGs heatmap

ht_BAL_UT_ISGs_BAL_order_meanAll <- Heatmap(analysis$rlogBALUT_FromMean[ISGs_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
                                    name = "log2FoldDiff\nfrom Mean All",
                                    cluster_rows = FALSE,
                                    show_column_dend = FALSE,
                                    column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx"),
                                    cluster_column_slices = FALSE,
#                                    bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                    clustering_distance_columns = "pearson",
                                    clustering_method_columns = "complete",
                                    show_column_names = FALSE,
                                    column_gap = unit(2,"mm"),
                                    col = heatscale4,
                                    row_names_gp = gpar(fontsize = 6),
                                    rect_gp = gpar(col = "black", lwd = 1),
                                    show_heatmap_legend = FALSE)
draw(ht_BAL_UT_ISGs_BAL_order_meanAll)

ht_BAL_UT_ISGs_BAL_order_meanAll_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs_BAL_order_meanAll, column_title = "BAL - ISGs mean All")), ncol = 1, nrow = 1)
ggsave(ht_BAL_UT_ISGs_BAL_order_meanAll_layout, filename = "BAL_ISGs_UT_BALorder_Heatmap_meanAll.pdf", units = "in", height = 3, width = 2.35, scale = 1)

## BAL UT meanAll IL6 heatmap

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll <- Heatmap(analysis$rlogBALUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        column_title = "BAL IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll,column_title = "BAL - IL6 JAK/STAT m All"))

ggsave(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot, filename = "BAL_IL6_UT_BALorder_Heatmap_meanAll.pdf", units = "in", height = 4, width = 2.35, scale = 1)
ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll

## BAL UT meanD0 ISGs heatmap

ht_BAL_UT_ISGs_BAL_order_meanD0 <- Heatmap(analysis$rlogBALUT_FromMeanDay0[ISGs_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
                                    name = "log2FoldDiff\nfrom Mean Day0",
                                    cluster_rows = FALSE,
                                    show_column_dend = FALSE,
                                    column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx"),
                                    cluster_column_slices = FALSE,
#                                    bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                    clustering_distance_columns = "pearson",
                                    clustering_method_columns = "complete",
                                    show_column_names = FALSE,
                                    column_gap = unit(2,"mm"),
                                    col = heatscale4,
                                    row_names_gp = gpar(fontsize = 6),
                                    rect_gp = gpar(col = "black", lwd = 1),
                                    show_heatmap_legend = FALSE)
draw(ht_BAL_UT_ISGs_BAL_order_meanD0)

ht_BAL_UT_ISGs_BAL_order_meanD0_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs_BAL_order_meanD0, column_title = "BAL - ISGs mean D0")), ncol = 1, nrow = 1)
ggsave(ht_BAL_UT_ISGs_BAL_order_meanD0_layout, filename = "BAL_ISGs_UT_BALorder_Heatmap_meanD0.pdf", units = "in", height = 3, width = 2.35, scale = 1)

## BAL UT meanD0 IL6 heatmap

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanD0 <- Heatmap(analysis$rlogBALUT_FromMeanDay0[IL6_JAK_STAT3_genes_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean Day0",
#        column_title = "BAL IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanD0_plot <- grid.grabExpr(draw(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanD0,column_title = "BAL - IL6 JAK/STAT m D0"))

ggsave(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanD0_plot, filename = "BAL_IL6_UT_BALorder_Heatmap_meanD0.pdf", units = "in", height = 4, width = 2.35, scale = 1)
ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanD0

## BAL UT meanBL ISGs heatmap

ht_BAL_UT_ISGs_BAL_order_meanBL <- Heatmap(analysis$rlogBALUT_FromMeanBL[ISGs_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
                                    name = "log2FoldDiff\nfrom Mean BL",
                                    cluster_rows = FALSE,
                                    show_column_dend = FALSE,
                                    column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx"),
                                    cluster_column_slices = FALSE,
#                                    bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                    clustering_distance_columns = "pearson",
                                    clustering_method_columns = "complete",
                                    show_column_names = FALSE,
                                    column_gap = unit(2,"mm"),
                                    col = heatscale4,
                                    row_names_gp = gpar(fontsize = 6),
                                    rect_gp = gpar(col = "black", lwd = 1),
                                    show_heatmap_legend = FALSE)
draw(ht_BAL_UT_ISGs_BAL_order_meanBL)

ht_BAL_UT_ISGs_BAL_order_meanBL_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs_BAL_order_meanBL, column_title = "BAL - ISGs mean BL")), ncol = 1, nrow = 1)
ggsave(ht_BAL_UT_ISGs_BAL_order_meanBL_layout, filename = "BAL_ISGs_UT_BALorder_Heatmap_meanBL.pdf", units = "in", height = 3, width = 2.35, scale = 1)

## BAL UT meanBL IL6 heatmap

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanBL <- Heatmap(analysis$rlogBALUT_FromMeanBL[IL6_JAK_STAT3_genes_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean BL",
#        column_title = "BAL IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanBL_plot <- grid.grabExpr(draw(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanBL,column_title = "BAL - IL6 JAK/STAT m BL"))

ggsave(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanBL_plot, filename = "BAL_IL6_UT_BALorder_Heatmap_meanBL.pdf", units = "in", height = 4, width = 2.35, scale = 1)
ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanBL

## PBMC UT meanAll ISGs heatmap

ht_PBMC_UT_ISGs_BAL_order_meanAll <- Heatmap(analysis$rlogPBMCUT_FromMean[ISGs_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        row_title = "Uninfected-IFNant",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_ISGs_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_UT_ISGs_BAL_order_meanAll,column_title = "PBMC - ISGs mean All"))

ggsave(ht_PBMC_UT_ISGs_BAL_order_meanAll_plot, filename = "PBMC_ISGs_UT_BALorder_Heatmap_meanAll.pdf", units = "in", height = 3, width = 3, scale = 1)
ht_PBMC_UT_ISGs_BAL_order_meanAll

## PBMC UT meanAll IL6 heatmap

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll <- Heatmap(analysis$rlogPBMCUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        column_title = "PBMC IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll,column_title = "PBMC - IL6 JAK/STAT mean All"))

ggsave(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot, filename = "PBMC_IL6_UT_BALorder_Heatmap_meanAll.pdf", units = "in", height = 4, width = 3, scale = 1)
ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll

## PBMC UT meanD0 ISGs heatmap

ht_PBMC_UT_ISGs_BAL_order_meanD0 <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[ISGs_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean Day0",
#        row_title = "Uninfected-IFNant",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-7d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_ISGs_BAL_order_meanD0_plot <- grid.grabExpr(draw(ht_PBMC_UT_ISGs_BAL_order_meanD0,column_title = "PBMC - ISGs mean D0"))

ggsave(ht_PBMC_UT_ISGs_BAL_order_meanD0_plot, filename = "PBMC_ISGs_UT_BALorder_Heatmap_meanD0.pdf", units = "in", height = 3, width = 3, scale = 1)
ht_PBMC_UT_ISGs_BAL_order_meanD0

## PBMC UT meanD0 IL6 heatmap

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanD0 <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean Day0",
#        column_title = "PBMC IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanD0_plot <- grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanD0,column_title = "PBMC - IL6 JAK/STAT mean D0"))

ggsave(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanD0_plot, filename = "PBMC_IL6_UT_BALorder_Heatmap_meanD0.pdf", units = "in", height = 4, width = 3, scale = 1)
ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanD0

## PBMC UT meanBL ISGs heatmap

ht_PBMC_UT_ISGs_BAL_order_meanBL <- Heatmap(analysis$rlogPBMCUT_FromMeanBL[ISGs_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean BL",
#        row_title = "Uninfected-IFNant",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_ISGs_BAL_order_meanBL_plot <- grid.grabExpr(draw(ht_PBMC_UT_ISGs_BAL_order_meanBL,column_title = "PBMC - ISGs mean BL"))

ggsave(ht_PBMC_UT_ISGs_BAL_order_meanBL_plot, filename = "PBMC_ISGs_UT_BALorder_Heatmap_meanBL.pdf", units = "in", height = 3, width = 3, scale = 1)
ht_PBMC_UT_ISGs_BAL_order_meanBL

## PBMC UT meanBL IL6 heatmap

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanBL <- Heatmap(analysis$rlogPBMCUT_FromMeanBL[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean BL",
#        column_title = "PBMC IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanBL_plot <- grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanBL,column_title = "PBMC - IL6 JAK/STAT mean BL"))

ggsave(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanBL_plot, filename = "PBMC_IL6_UT_BALorder_Heatmap_meanBL.pdf", units = "in", height = 4, width = 3, scale = 1)
ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanBL



```


### For Rebutal

```{r}

## BAL UT meanAll ISGs heatmap

ht_BAL_UT_ISGs_BAL_order_meanAll <- Heatmap(analysis$rlogBALUT_FromMean[ISGs_BAL_order[-5],analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
                                    name = "log2FoldDiff\nfrom Mean All",
                                    cluster_rows = FALSE,
                                    show_column_dend = FALSE,
                                    column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx"),
                                    cluster_column_slices = FALSE,
#                                    bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                    clustering_distance_columns = "pearson",
                                    clustering_method_columns = "complete",
                                    show_column_names = FALSE,
                                    column_gap = unit(2,"mm"),
                                    col = heatscale4,
                                    row_names_gp = gpar(fontsize = 6),
                                    rect_gp = gpar(col = "black", lwd = 1),
                                    show_heatmap_legend = FALSE)
draw(ht_BAL_UT_ISGs_BAL_order_meanAll)
ht_BAL_UT_ISGs_BAL_order_meanAll

ht_BAL_UT_ISGs_BAL_order_meanAll_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs_BAL_order_meanAll, column_title = "BAL - ISGs")), ncol = 1, nrow = 1)
ggsave(ht_BAL_UT_ISGs_BAL_order_meanAll_layout, filename = "BAL_ISGs_UT_BALorder_Heatmap_meanAll.png", units = "in", height = 3, width = 2.35, scale = 1)

as.data.frame(analysis$rlogBALUT_FromMean[ISGs_BAL_order[-5],analysis$rldBALUT$Treatment == "Uninfected-IFNant"]) %>% rename_with(~paste(as.character(recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx")),.x)) %>%
   rownames_to_column(var = "gene") %>%
   write.xlsx(file = "BAL_ISGs_UT_BALorder_Heatmap_meanAll_table.xlsx", colNames = TRUE)


## BAL UT meanAll IL6 heatmap

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll <- Heatmap(analysis$rlogBALUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        column_title = "BAL IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll,column_title = "BAL - IL6 JAK/STAT"))

ggsave(ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot, filename = "BAL_IL6_UT_BALorder_Heatmap_meanAll.png", units = "in", height = 4, width = 2.35, scale = 1)
ht_BAL_UT_IL6_JAK_STAT3_BAL_order_meanAll

as.data.frame(analysis$rlogBALUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "BAL_IL6_UT_BALorder_Heatmap_meanAll_table.xlsx", colNames = TRUE)

## PBMC UT meanAll ISGs heatmap

ht_PBMC_UT_ISGs_BAL_order_meanAll <- Heatmap(analysis$rlogPBMCUT_FromMean[ISGs_BAL_order[-5],analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        row_title = "Uninfected-IFNant",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_ISGs_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_UT_ISGs_BAL_order_meanAll,column_title = "PBMC - ISGs"))

ggsave(ht_PBMC_UT_ISGs_BAL_order_meanAll_plot, filename = "PBMC_ISGs_UT_BALorder_Heatmap_meanAll.png", units = "in", height = 3, width = 3, scale = 1)
ht_PBMC_UT_ISGs_BAL_order_meanAll

as.data.frame(analysis$rlogPBMCUT_FromMean[ISGs_BAL_order[-5],analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]) %>% rename_with(~paste(as.character(recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx")),.x)) %>%
   rownames_to_column(var = "gene") %>%
   write.xlsx(file = "PBMC_ISGs_UT_BALorder_Heatmap_meanAll_table.xlsx", colNames = TRUE)


## PBMC UT meanAll IL6 heatmap

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll <- Heatmap(analysis$rlogPBMCUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean All",
#        column_title = "PBMC IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll,column_title = "PBMC - IL6 JAK/STAT"))

ggsave(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll_plot, filename = "PBMC_IL6_UT_BALorder_Heatmap_meanAll.png", units = "in", height = 4, width = 3, scale = 1)
ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_meanAll

as.data.frame(analysis$rlogPBMCUT_FromMean[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "PBMC_IL6_UT_BALorder_Heatmap_meanAll_table.xlsx", colNames = TRUE)

```


## Rebuttal reorg supplemental

```{r}

ht_PBMC_Ctrl_IL6_JAK_STAT3_meanAll <- Heatmap(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "Untreated"],
                                      name = "log2FoldDiff\nfrom Mean",
                                      row_title = "Untreated",
                                      cluster_rows = FALSE,
                                     column_split = recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi"),
                                         column_title_gp = gpar(fontsize = 8),
                                     cluster_column_slices = FALSE,
#                                      bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))),
                                      clustering_distance_columns = "pearson",
                                      clustering_method_columns = "complete",
                                      show_column_dend = FALSE,
                                      show_column_names = FALSE,
                                      col = heatscale4,
                                      row_title_gp = gpar(col = "red"),
                                      row_names_gp = gpar(fontsize = 6),
                                      rect_gp = gpar(col = "black", lwd = 1),
show_heatmap_legend = FALSE)
#draw(ht_PBMC_Ctrl_IL6_JAK_STAT3)


ht_PBMC_IFNant_IL6_JAK_STAT3_meanAll <- Heatmap(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "IFNant"],
                                        name = "log2FoldDiff\nfrom Mean",
                                        row_title = "IFNmod",
                                        cluster_rows = FALSE,
                                        column_split = recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi"),
                                        column_title = NULL,
                                        cluster_column_slices = FALSE,
                                        show_column_dend = FALSE,
#                                        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                        show_column_names = FALSE,
                                        col = heatscale4,
                                        row_title_gp = gpar(col = "blue"),
                                        row_names_gp = gpar(fontsize = 6),
                                        rect_gp = gpar(col = "black", lwd = 1),
show_heatmap_legend = FALSE)
#draw(ht_PBMC_IFNant_IL6_JAK_STAT3, column_title = "IL6_JAK_STAT3 SIGNALING in PBMC")

ht_PBMC_IFNant_IL6_JAK_STAT3_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_IFNant_IL6_JAK_STAT3_meanAll))
ht_PBMC_Ctrl_IL6_JAK_STAT3_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_Ctrl_IL6_JAK_STAT3_meanAll, column_title = "PBMC  - IL6_JAK_STAT3 SIGNALING"))
ht_PBMC_IL6_JAK_STAT3_meanAll_layout <- ggarrange(ht_PBMC_Ctrl_IL6_JAK_STAT3_meanAll_plot, ht_PBMC_IFNant_IL6_JAK_STAT3_meanAll_plot,  ncol = 1, nrow = 2, heights = c(1.15,0.85))
ggsave(ht_PBMC_IL6_JAK_STAT3_meanAll_layout, filename = "IL6_JAK_STAT3_SIGNALING_PBMC_Heatmap_meanAll.pdf", units = "in", height = 4.75, width = 5, scale = 1)
ggsave(ht_PBMC_IL6_JAK_STAT3_meanAll_layout, filename = "IL6_JAK_STAT3_SIGNALING_PBMC_Heatmap_meanAll.png", units = "in", height = 4.75, width = 5, scale = 1)

# as.data.frame(analysis$rlogBALUT_FromMean[ISGs_BAL_order[-5],analysis$rldBALUT$Treatment == "Uninfected-IFNant"]) %>% rename_with(~paste(as.character(recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx")),.x)) %>%
#    rownames_to_column(var = "gene") %>%
#    write.xlsx(file = "BAL_ISGs_UT_BALorder_Heatmap_meanAll_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "IL6_JAK_STAT3_SIGNALING_PBMC_Heatmap_meanAll_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% Bari_IL6_JAK_STAT3,analysis$rldPBMCp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "IL6_JAK_STAT3_SIGNALING_PBMC_Heatmap_meanAll_IFNmod.xlsx", colNames = TRUE)


```


```{r}

ht_PBMC_Ctrl_ISGs_meanAll <- Heatmap(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% ISGList[-18],analysis$rldPBMCp12$Treatment == "Untreated"],
                                      name = "log2FoldDiff\nfrom Mean",
                                      row_title = "Untreated",
                                      cluster_rows = FALSE,
                                     column_split = recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi"),
                                         column_title_gp = gpar(fontsize = 8),
                                     cluster_column_slices = FALSE,
#                                      bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))),
                                      clustering_distance_columns = "pearson",
                                      clustering_method_columns = "complete",
                                      show_column_dend = FALSE,
                                      show_column_names = FALSE,
                                      col = heatscale4,
                                      row_title_gp = gpar(col = "red"),
                                      row_names_gp = gpar(fontsize = 6),
                                      rect_gp = gpar(col = "black", lwd = 1),
show_heatmap_legend = FALSE)
#draw(ht_PBMC_Ctrl_ISGs_meanAll)


ht_PBMC_IFNant_ISGs_meanAll <- Heatmap(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% ISGList[-18],analysis$rldPBMCp12$Treatment == "IFNant"],
                                        name = "log2FoldDiff\nfrom Mean",
                                        row_title = "IFNmod",
                                        cluster_rows = FALSE,
                                        column_split = recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi"),
                                        column_title = NULL,
                                        cluster_column_slices = FALSE,
                                        show_column_dend = FALSE,
#                                        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                        show_column_names = FALSE,
                                        col = heatscale4,
                                        row_title_gp = gpar(col = "blue"),
                                        row_names_gp = gpar(fontsize = 6),
                                        rect_gp = gpar(col = "black", lwd = 1),
show_heatmap_legend = FALSE)
#draw(ht_PBMC_IFNant_ISGs_meanAll, column_title = "PBMCs - ISGs")

ht_PBMC_IFNant_ISGs_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_IFNant_ISGs_meanAll))
ht_PBMC_Ctrl_ISGs_meanAll_plot <- grid.grabExpr(draw(ht_PBMC_Ctrl_ISGs_meanAll, column_title = "PBMCs - ISGs"))
ht_PBMC_ISGs_meanAll_layout <- ggarrange(ht_PBMC_Ctrl_ISGs_meanAll_plot, ht_PBMC_IFNant_ISGs_meanAll_plot,  ncol = 1, nrow = 2, heights = c(1.15,0.85))
ggsave(ht_PBMC_ISGs_meanAll_layout, filename = "ISGs_PBMC_Heatmap_meanAll.pdf", units = "in", height = 3.75, width = 5, scale = 1)
ggsave(ht_PBMC_ISGs_meanAll_layout, filename = "ISGs_PBMC_Heatmap_meanAll.png", units = "in", height = 3.75, width = 5, scale = 1)


as.data.frame(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% ISGList[-18],analysis$rldPBMCp12$Treatment == "Untreated"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "ISGs_PBMC_Heatmap_meanAll_Untreated_table.xlsx", colNames = TRUE)

as.data.frame(analysis$rlogPBMCp12_FromMean[rownames(analysis$rlogPBMCp12_FromMean) %in% ISGList[-18],analysis$rldPBMCp12$Treatment == "IFNant"]) %>%
  rename_with(~paste(as.character(recode_factor(analysis$rldPBMCp12[,analysis$rldPBMCp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day0 = "0 dpi", Day1 = "1 dpi", Day2 = "2 dpi", Day4 = "4 dpi", Day5 = "5 dpi", Day7 = "7 dpi")),.x)) %>%
  rownames_to_column(var = "gene") %>%
  write.xlsx(file = "ISGs_PBMC_Heatmap_meanAll_IFNmod_table.xlsx", colNames = TRUE)


```




```{r fig.height=5, fig.width=11}

ht_BAL_UT_ISGs_BAL_order <- Heatmap(analysis$rlogBALUT_FromMeanDay0[ISGs_BAL_order,analysis$rldBALUT$Treatment == "Uninfected-IFNant"],
                                    name = "log2FoldDiff\nfrom Mean Day0",
                                    cluster_rows = FALSE,
                                    show_column_dend = FALSE,
                                    column_split = recode_factor(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-3d tx", Day0 = "1d tx", Day2 = "3d tx"),
                                    cluster_column_slices = FALSE,
#                                    bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALUT[,analysis$rldBALUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
                                    clustering_distance_columns = "pearson",
                                    clustering_method_columns = "complete",
                                    show_column_names = FALSE,
                                    column_gap = unit(2,"mm"),
                                    col = heatscale4,
                                    row_names_gp = gpar(fontsize = 6),
                                    rect_gp = gpar(col = "black", lwd = 1),
                                    show_heatmap_legend = FALSE)
draw(ht_BAL_UT_ISGs_BAL_order)

ht_BAL_UT_ISGs_BAL_order_layout <- ggarrange(grid.grabExpr(draw(ht_BAL_UT_ISGs_BAL_order, column_title = "BAL - ISGs")), ncol = 1, nrow = 1)
ggsave(ht_BAL_UT_ISGs_BAL_order_layout, filename = "BAL_ISGs_UT_BALorder_Heatmap.pdf", units = "in", height = 3, width = 2.35, scale = 1)


```




```{r}

#Fig 1E PBMC IL6 signalling/inflammatory

IL6_JAK_STAT3_genes_BAL_order <- c("CSF3R","IL1R2","IL1R1","OSMR","CXCL3","CCL7","LTB","CD38","IL12RB1","IL18R1","SOCS3","IFNGR2","IFNAR1","CSF2","CD14","PIK3R5","IL4R","CXCL13","CXCL9","IFNGR1","CD36","TLR2","CCR1","HMOX1")

# Heatmap(analysis$rlogBALp12_FromMean[t_activation_IFNmod_order,analysis$rldBALp12$Treatment == "IFNant"],name = "log2FoldDiff\nfrom Mean", row_title = "IFNmod", row_title_gp = gpar(fontsize = 12, col = "#003C5F"), cluster_rows = FALSE, show_column_dend = FALSE, column_split = recode_factor(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$Timepoint, BL = "baseline", Day2 = "2 dpi", Day4 = "4 dpi", Day7 = "7 dpi"), cluster_column_slices = FALSE,
# #                                                     bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldBALp12[,analysis$rldBALp12$Treatment == "Untreated"]$SubjectID, gp = gpar(fontsize = 7))), 
#                                                      clustering_distance_columns = "pearson", clustering_method_columns = "complete",show_column_names = FALSE, column_gap = unit(2,"mm"), column_title_gp = gpar(fontsize = 6), col = heatscale4, row_names_gp = gpar(fontsize = 6), rect_gp = gpar(col = "black", lwd = 1), show_heatmap_legend = FALSE)
# 


ht_PBMC_UT_IL6_JAK_STAT3_BAL_order <- Heatmap(analysis$rlogPBMCUT_FromMeanDay0[IL6_JAK_STAT3_genes_BAL_order,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"],
        name = "log2FoldDiff\nfrom Mean Day0",
#        column_title = "PBMC IL6 signalling/inflamatory",
#        row_title = "Uninfected-IFNmod",
        cluster_rows = FALSE,
        show_column_dend = FALSE,
        column_split = recode_factor(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$Timepoint, BL = "-7d tx", Day0 = "1d tx", Day1 = "2d tx", Day2 = "3d tx"),
        cluster_column_slices = FALSE,
#        bottom_annotation = HeatmapAnnotation(SubID = anno_text(analysis$rldPBMCUT[,analysis$rldPBMCUT$Treatment == "Uninfected-IFNant"]$SubjectID, gp = gpar(fontsize = 7))),
        clustering_distance_columns = "pearson",
        clustering_method_columns = "complete",
        show_column_names = FALSE,
        column_gap = unit(2,"mm"),
        col = heatscale4,
        row_names_gp = gpar(fontsize = 6),
        rect_gp = gpar(col = "black", lwd = 1),
        show_heatmap_legend = FALSE)

ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_plot <- grid.grabExpr(draw(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order,column_title = "PBMC IL6 signalling/inflamatory"))

ggsave(ht_PBMC_UT_IL6_JAK_STAT3_BAL_order_plot, filename = "PBMC_IL6_Fig_Heatmaps_small.pdf", units = "in", height = 4, width = 3, scale = 1)
ht_PBMC_UT_IL6_JAK_STAT3_BAL_order

```



## GSEA Enrichment Plots

```{r}

enrich_TNFA <- One.GSEA.Enrich.Plot2(dir1 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",dir2 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",geneset = "HALLMARK_TNFA_SIGNALING_VIA_NFKB",color1 = "#005477",color2 = "#FA8072",mainTitle = "TNFA",phen1="Day2",phen2 = "Baseline",y_outreach = 0.8,label1="IFNmod",label2 = "Untreated")

ggsave(enrich_TNFA, filename = "TNFA_enrichmentPlot.pdf", units = "in", width = 8, height = 8, scale = 1)

enrich_IL6JAKSTAT <- One.GSEA.Enrich.Plot2(dir1 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",dir2 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",geneset = "HALLMARK_IL6_JAK_STAT3_SIGNALING",color1 = "#005477",color2 = "#FA8072",mainTitle = "IL6_JAK_STAT3",phen1="Day2",phen2 = "Baseline",y_outreach = 0.8,label1="IFNmod",label2 = "Untreated")

ggsave(enrich_IL6JAKSTAT, filename = "IL6_JAK_STAT3_BAL_enrichmentPlot.pdf", units = "in", width = 8, height = 8, scale = 1)

enrich_NEUTDEGRAN <- One.GSEA.Enrich.Plot2(dir1 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",dir2 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",geneset = "REACTOME_NEUTROPHIL_DEGRANULATION",color1 = "#005477",color2 = "#FA8072",mainTitle = "NEUTROPHIL_DEGRANULATION",phen1="Day2",phen2 = "Baseline",y_outreach = 0.5,label1="IFNmod",label2 = "Untreated")

ggsave(enrich_NEUTDEGRAN, filename = "NEUTDEGRAN_BAL_enrichmentPlot.pdf", units = "in", width = 8, height = 8, scale = 1)

enrich_ISGs <- One.GSEA.Enrich.Plot(dir1 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2vsBL_IFNant_BAL_shortlist_pGS.Gsea.1651713870797/",dir2 = "/yerkes-cifs/runs/Analysis/COVten_Bulk/p21094_COVten_Bulk_Analysis/GSEA/Day2VsBL_Untreated_BAL_shortlist_pGS.Gsea.1636386908828/",geneset = "ISG",color1 = "#005477",color2 = "#FA8072",mainTitle = "ISGs",phen1="Day2",phen2 = "Baseline",y_outreach = 1,label1="IFNmod",label2 = "Untreated")

ggsave(enrich_ISGs, filename = "ISGs_BAL_enrichmentPlot.pdf", units = "in", width = 8, height = 8, scale = 1)




```

# Summary Sample Table

```{r}

phase1AcolData <- as.data.frame(SummarizedExperiment::colData(analysis$dds))
phase1AsampleTable <- mutate(phase1AcolData, Infection = "Infected",
                             Treatment = factor(if_else(Treatment == "Untreated","Control","IFNmod"), levels = c("Control","IFNmod"))) %>%
  select(c("SampleID","SubjectID","Source","Timepoint","Infection","Treatment"))
phase1Arawcounts <- DESeq2::counts(analysis$dds)


phase1BcolData <- as.data.frame(SummarizedExperiment::colData(analysis$dds2))
phase1BsampleTable <- mutate(phase1BcolData, Infection = "Infected",
                             Treatment = factor(if_else(Treatment == "Untreated","Control","IFNmod"), levels = c("Control","IFNmod"))) %>%
  select(c("SampleID","SubjectID","Source","Timepoint","Infection","Treatment"))
phase1Brawcounts <- DESeq2::counts(analysis$dds2)


phase0colData <- as.data.frame(SummarizedExperiment::colData(analysis$ddsUT))
phase0sampleTable <- mutate(phase0colData,
                            Source = factor(Source, levels = c("BAL","PBMC")),
                            Timepoint = factor(Timepoint, levels = c("BL","Day0","Day1","Day2")),
                            Infection = "Not-Infected",
                            Treatment = factor(if_else(Treatment == "Uninfected-IFNant","IFNmod",""), levels = c("Control","IFNmod"))) %>%
  select(c("SampleID","SubjectID","Source","Timepoint","Infection","Treatment"))
phase0rawcounts <- DESeq2::counts(analysis$ddsUT)


summarySampleTable <- rbind(phase1AsampleTable, phase1BsampleTable) %>%
  rbind(phase0sampleTable)
write_csv(summarySampleTable, path = "COVTen_sampleTable.csv", col_names = TRUE)

summaryRawCounts <- cbind(phase1Arawcounts, phase1Brawcounts) %>%
  cbind(phase0rawcounts)
write_csv(as.data.frame(summaryRawCounts)%>%rownames_to_column(var = "EnsemblID"),
          path = "COVTen_Raw_Counts.csv",
          col_names = TRUE)

```

# Supplemental Plots 2023


```{r}

plotGenes <- c("CLEC4C","IFNA6","IFNA8","IFNA10","IFNA13","IFNA14")

plotList <- lapply(plotGenes, function(x) {plotCountsGKT(analysis$ddsBALp12[,analysis$ddsBALp12$Timepoint == "Day2"],x,intgroup = c("Timepoint","Treatment")) + geom_jitter(aes(color="Treatment"), width = 0.1) + scale_color_manual(labels = c("Control","INFant"), values = c("#F73019","#003C5F")) + theme_bw()})

ggsave(ggarrange(plotlist = plotList), filename = "CLEC4C_IFNA_Plots.pdf", units = "in", width = 11, height = 8.5, scale = 1)

p_CLEC4C <-plotCountsGKT(analysis$ddsBALp12,"CLEC4C",intgroup = c("Timepoint","Treatment")) + geom_jitter(width = 0.1)

p_IFNA8 <-plotCountsGKT(analysis$ddsBALp12,"IFNA8",intgroup = c("Timepoint","Treatment")) + geom_jitter(width = 0.1)

ggarrange(p_CLEC4C,p_IFNA8)

```


